From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:06:34 -0700
Subject: [PATCH] Build changes


diff --git a/build.gradle b/build.gradle
index e70dcca1a7829f0df5768e90c25dd66cda7c0e36..6fac8a1b3befef2179dc874856085f362cf473b4 100644
--- a/build.gradle
+++ b/build.gradle
@@ -25,8 +25,10 @@ archivesBaseName = project.name
 def baseVersion = '0.10'
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-	version = baseVersion + '.' + ENV.BUILD_NUMBER
+// quiet start
+if (ENV.GITHUB_RUN_NUMBER) {
+	version = baseVersion + '.' + ENV.GITHUB_RUN_NUMBER
+	// quiet end
 } else {
 	version = baseVersion + '.local'
 }
@@ -148,7 +150,7 @@ codenarc {
 gradlePlugin {
 	plugins {
 		fabricLoom {
-			id = 'fabric-loom'
+			id = 'quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.bootstrap.LoomGradlePluginBootstrap'
 		}
 	}
@@ -224,8 +226,10 @@ publishing {
 
 		// Manually crate the plugin marker for snapshot versions
 		snapshotPlugin(MavenPublication) { publication ->
-			groupId 'fabric-loom'
-			artifactId 'fabric-loom.gradle.plugin'
+			// quiet start
+			groupId 'quiet-fabric-loom'
+			artifactId 'quiet-fabric-loom.gradle.plugin'
+			// quiet end
 			version baseVersion + '-SNAPSHOT'
 
 			pom.withXml({
@@ -237,7 +241,7 @@ publishing {
 				Node groupId = dependency.appendChild(document.createElement('groupId'))
 				groupId.setTextContent('net.fabricmc')
 				Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-				artifactId.setTextContent('fabric-loom')
+				artifactId.setTextContent('quiet-fabric-loom') // quiet
 				Node version = dependency.appendChild(document.createElement('version'))
 				version.setTextContent(baseVersion + '-SNAPSHOT')
 			})
@@ -245,13 +249,13 @@ publishing {
 	}
 	repositories {
 		maven {
-			if (ENV.MAVEN_URL) {
-				url ENV.MAVEN_URL
-				credentials {
-					username ENV.MAVEN_USERNAME
-					password ENV.MAVEN_PASSWORD
-				}
+			// quiet start
+			url("https://repo.jpenilla.xyz/snapshots/")
+			credentials {
+				username = System.getenv("JMP_REPO_USER")
+				password = System.getenv("JMP_REPO_PASSWORD")
 			}
+			// quiet end
 		}
 	}
 }
diff --git a/settings.gradle b/settings.gradle
index 885070a6d762361cb5edf6ce1f85a72e5b353407..3f0777d21b41a8c7aa32f04c7e32fe6f4d171916 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,3 +1,3 @@
-rootProject.name = name
+rootProject.name = "quiet-fabric-loom" // quiet
 
 include "bootstrap"
\ No newline at end of file
