From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:06:34 -0700
Subject: [PATCH] Build changes & rebrand


diff --git a/.gitattributes b/.gitattributes
index 36452a5b0cba931fd34b8179fc218c81b8208f03..9c62f78cffa160885fd599c0a4bf9e50c089611a 100644
--- a/.gitattributes
+++ b/.gitattributes
@@ -36,3 +36,9 @@
 *.png           binary
 *.so            binary
 *.war           binary
+
+#
+# https://help.github.com/articles/dealing-with-line-endings/
+#
+# These are explicitly windows files and should use crlf
+*.bat           text eol=crlf
diff --git a/build.gradle b/build.gradle
index 10a6e0c0dc68d863684b37e904396fb7b1cab39f..cf94d6845a1017b91ac4cd3bbdb5506cee5a3456 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,12 +5,13 @@ plugins {
 	id 'idea'
 	id 'eclipse'
 	id 'groovy'
-	id 'checkstyle'
+	// id 'checkstyle' // quiet - remove checkstyle
 	id 'jacoco'
 	id 'codenarc'
 	id "org.jetbrains.kotlin.jvm" version "1.6.10" // Must match the version included with gradle.
 	id "com.diffplug.spotless" version "6.13.0"
 	id "org.gradle.test-retry" version "1.5.1"
+	id "com.adarshr.test-logger" version "3.2.0" // quiet
 }
 
 tasks.withType(JavaCompile).configureEach {
@@ -28,8 +29,10 @@ archivesBaseName = project.name
 def baseVersion = '1.1'
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-	version = baseVersion + '.' + ENV.BUILD_NUMBER
+// quiet start
+if (ENV.GITHUB_RUN_NUMBER) {
+	version = baseVersion + '.' + ENV.GITHUB_RUN_NUMBER
+	// quiet end
 } else {
 	version = baseVersion + '.local'
 }
@@ -158,10 +161,13 @@ spotless {
 	}
 }
 
-checkstyle {
-	configFile = file('checkstyle.xml')
-	toolVersion = '10.6.0'
+// quiet - remove checkstyle
+
+// quiet start
+testlogger {
+	theme = com.adarshr.gradle.testlogger.theme.ThemeType.MOCHA_PARALLEL
 }
+// quiet end
 
 codenarc {
 	toolVersion = "3.2.0"
@@ -171,7 +177,7 @@ codenarc {
 gradlePlugin {
 	plugins {
 		fabricLoom {
-			id = 'fabric-loom'
+			id = 'quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.bootstrap.LoomGradlePluginBootstrap'
 		}
 	}
@@ -227,8 +233,10 @@ publishing {
 
 		// Manually crate the plugin marker for snapshot versions
 		snapshotPlugin(MavenPublication) { publication ->
-			groupId 'fabric-loom'
-			artifactId 'fabric-loom.gradle.plugin'
+			// quiet start
+			groupId 'quiet-fabric-loom'
+			artifactId 'quiet-fabric-loom.gradle.plugin'
+			// quiet end
 			version baseVersion + '-SNAPSHOT'
 
 			pom.withXml({
@@ -240,7 +248,7 @@ publishing {
 				Node groupId = dependency.appendChild(document.createElement('groupId'))
 				groupId.setTextContent('net.fabricmc')
 				Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-				artifactId.setTextContent('fabric-loom')
+				artifactId.setTextContent('quiet-fabric-loom') // quiet
 				Node version = dependency.appendChild(document.createElement('version'))
 				version.setTextContent(baseVersion + '-SNAPSHOT')
 			})
@@ -248,13 +256,13 @@ publishing {
 	}
 	repositories {
 		maven {
-			if (ENV.MAVEN_URL) {
-				url ENV.MAVEN_URL
-				credentials {
-					username ENV.MAVEN_USERNAME
-					password ENV.MAVEN_PASSWORD
-				}
+			// quiet start
+			url("https://repo.jpenilla.xyz/snapshots/")
+			credentials {
+				username = System.getenv("JMP_REPO_USER")
+				password = System.getenv("JMP_REPO_PASSWORD")
 			}
+			// quiet end
 		}
 	}
 }
diff --git a/settings.gradle b/settings.gradle
index 885070a6d762361cb5edf6ce1f85a72e5b353407..3f0777d21b41a8c7aa32f04c7e32fe6f4d171916 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,3 +1,3 @@
-rootProject.name = name
+rootProject.name = "quiet-fabric-loom" // quiet
 
 include "bootstrap"
\ No newline at end of file
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index a75b67fcbd76a74b9e6f4786dd5bc5e19690790b..d0233a19d7bbe7b1152d34f18d9e949ab8a9c80c 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -62,7 +62,7 @@ public class LoomGradlePlugin implements BootstrappedPlugin {
 	}
 
 	public void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quiet Fabric Loom: " + LOOM_VERSION); // quiet
 		LibraryLocationLogger.logLibraryVersions();
 
 		// Apply default plugins
diff --git a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
index 7fd5d15941dddd97c2631c26b243b7f320d9337f..ff5fdcfd4d229e0cb0418b9ff89a4d573039e40a 100644
--- a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
@@ -55,7 +55,7 @@ public final class GradleUtils {
 	}
 
 	public static boolean isLoomProject(Project project) {
-		return project.getPluginManager().hasPlugin("fabric-loom");
+		return project.getPluginManager().hasPlugin("quiet-fabric-loom"); // quiet
 	}
 
 	public static Provider<Boolean> getBooleanPropertyProvider(Project project, String key) {
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
index d67a1159a166972a56e03d97c94e3217691c65f8..6c056a996155748eb33b05dc336faeaf076b8a6f 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
@@ -53,6 +53,7 @@ class FabricAPITest extends Specification implements GradleProjectTestTrait {
 
 			// Set the version to something constant
 			gradle.buildGradle.text = gradle.buildGradle.text.replace('project.version + "+" + (ENV.GITHUB_RUN_NUMBER ? "" : "local-") + getBranch()', "\"$API_VERSION\"")
+					.replace("fabric-loom", "quiet-fabric-loom") // quiet
 
 			def server = ServerRunner.create(gradle.projectDir, "1.19.3")
 										.withMod(gradle.getOutputFile("fabric-api-${API_VERSION}.jar"))
diff --git a/src/test/resources/projects/accesswidener/build.gradle b/src/test/resources/projects/accesswidener/build.gradle
index 9fa3cc36ef23ee7f9a03120624596c6fe1788de4..6b033c7b0002832af062c1acb867b3304ff88218 100644
--- a/src/test/resources/projects/accesswidener/build.gradle
+++ b/src/test/resources/projects/accesswidener/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index 993ac667071e4e3e2b1a8301121e817950663d61..b38fc44fb5f10c691e2d01b6d1b1e97575ec5f38 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
index c7b6ef6c0a20c1d4b2936dcbdf222f201be41ccc..5cada09eb9648958f9748ff77046bf69f2d37f41 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
index 6472ec7e63ab88f17b626c68393faca58042a8d2..d09f8dba9f427000af38974497c434753a810b72 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
index c05e369995a4d8c5479de657d6c8ab52a6cba940..311b16a654c8e51f2eeaf3a090f883ba2f020f42 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
@@ -1,11 +1,11 @@
 pluginManagement {
     plugins {
-        id 'fabric-loom'
+        id 'quiet-fabric-loom' // quiet
     }
 }
 
 plugins {
-    id 'fabric-loom'
+    id 'quiet-fabric-loom' // quiet
 }
 
 dependencyResolutionManagement {
diff --git a/src/test/resources/projects/interfaceInjection/build.gradle b/src/test/resources/projects/interfaceInjection/build.gradle
index 2396f3c34efc5d82e7af392cfa52ef420e5735fd..90e56515cfd38173aadad3e1eb6bdd3d49cb83af 100644
--- a/src/test/resources/projects/interfaceInjection/build.gradle
+++ b/src/test/resources/projects/interfaceInjection/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/java16/build.gradle b/src/test/resources/projects/java16/build.gradle
index 45acf714efb54893e7867807d3bed47aa1a58223..93fe880c150bf047a93ab5d3ccb39954b859904d 100644
--- a/src/test/resources/projects/java16/build.gradle
+++ b/src/test/resources/projects/java16/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/kotlin/build.gradle.kts b/src/test/resources/projects/kotlin/build.gradle.kts
index 7f2e0f5d28bd402223319574cf804cc7ddeabc62..ad6c5ecbc9ab0a39d2f3151a6cbfdb1550cff52a 100644
--- a/src/test/resources/projects/kotlin/build.gradle.kts
+++ b/src/test/resources/projects/kotlin/build.gradle.kts
@@ -3,7 +3,7 @@ import java.util.Properties
 plugins {
 	kotlin("jvm") version "1.7.22"
 	kotlin("plugin.serialization") version "1.7.22"
-	id("fabric-loom")
+	id("quiet-fabric-loom") // quiet
 	`maven-publish`
 }
 
diff --git a/src/test/resources/projects/legacy/build.gradle b/src/test/resources/projects/legacy/build.gradle
index 3b460c49aaa95a8b98cff43df820b257498ad668..5761c8bedcb5493791e108044d5b3da4974f7a31 100644
--- a/src/test/resources/projects/legacy/build.gradle
+++ b/src/test/resources/projects/legacy/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 sourceCompatibility = JavaVersion.VERSION_1_8
diff --git a/src/test/resources/projects/localFileDependency/build.gradle b/src/test/resources/projects/localFileDependency/build.gradle
index 811d945a9a903d2e43e949de2e2fea4f99882edd..afa0c16e6c39f02e3b7dd0f6421ba7174874ec92 100644
--- a/src/test/resources/projects/localFileDependency/build.gradle
+++ b/src/test/resources/projects/localFileDependency/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/localRuntime/build.gradle b/src/test/resources/projects/localRuntime/build.gradle
index 18f4f0db4b857098a782cfa548a2954c44f363dc..80064373797d6f4de27cf041bc99d0d6a38ed651 100644
--- a/src/test/resources/projects/localRuntime/build.gradle
+++ b/src/test/resources/projects/localRuntime/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/maven/build.gradle b/src/test/resources/projects/maven/build.gradle
index 2c3e3786ffb233a2d4e355d21918552a067bf2da..44c0b7dd844e7545a0e0665c867b8378ed83e4c4 100644
--- a/src/test/resources/projects/maven/build.gradle
+++ b/src/test/resources/projects/maven/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mavenLibrary/build.gradle b/src/test/resources/projects/mavenLibrary/build.gradle
index 0001e2b7dd835ce6e1ef90415bb373563ec8e823..3279b8b352ba1cc4aa8baa186245947bc6cd432f 100644
--- a/src/test/resources/projects/mavenLibrary/build.gradle
+++ b/src/test/resources/projects/mavenLibrary/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/minimalBase/build.gradle b/src/test/resources/projects/minimalBase/build.gradle
index 8fbea237ed6254ea326fecc3c6db1a732746a6e0..7c5e9db47caaf722348db871c5503847fc0156bc 100644
--- a/src/test/resources/projects/minimalBase/build.gradle
+++ b/src/test/resources/projects/minimalBase/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApAutoRefmap/build.gradle b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
index 610d042d1a47cb4e3a82aad6e147b87248671333..9a912796ff75cc100f33937220eea76522a686df 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/build.gradle
+++ b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'com.github.johnrengelman.shadow' version '7.0.0'
 	id 'maven-publish'
 }
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index b21803ea86251da5162273226c5b38e3e4907f3f..73e08d3fbe0eb2d57fdbc0035746841398e52c1e 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/modJavadoc/build.gradle b/src/test/resources/projects/modJavadoc/build.gradle
index 52f1fc39f472e431795ddfe94cd85c6a01eade48..b374f3b02da2f1952df09e75759a2405d10c7c68 100644
--- a/src/test/resources/projects/modJavadoc/build.gradle
+++ b/src/test/resources/projects/modJavadoc/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mojangMappings/build.gradle b/src/test/resources/projects/mojangMappings/build.gradle
index 895bdb752a2aec465a2d430814229a1b2d35b0ee..2b748808c5d9f21a55cce1612b4a32b692569ae7 100644
--- a/src/test/resources/projects/mojangMappings/build.gradle
+++ b/src/test/resources/projects/mojangMappings/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/multiproject/build.gradle b/src/test/resources/projects/multiproject/build.gradle
index 62858e2f46a1c7d4e7cbc5fe7337bb90961c8490..3f8cbb87f9b640a5b4c0843793fa2e44aae10c3d 100644
--- a/src/test/resources/projects/multiproject/build.gradle
+++ b/src/test/resources/projects/multiproject/build.gradle
@@ -1,11 +1,11 @@
 plugins {
 	id "java"
 	id "java-library"
-	id 'fabric-loom' apply false
+	id 'quiet-fabric-loom' apply false // quiet
 }
 
 allprojects {
-	apply plugin: "fabric-loom"
+	apply plugin: "quiet-fabric-loom" // quiet
 
 	version = "1.0.0"
 
diff --git a/src/test/resources/projects/parchment/build.gradle b/src/test/resources/projects/parchment/build.gradle
index 5fee81c5d1e07de0dfcf615795ec818a8e4705e2..b2f68f83de8a72fd9be20681c788eb1efc6befb2 100644
--- a/src/test/resources/projects/parchment/build.gradle
+++ b/src/test/resources/projects/parchment/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/reproducible/build.gradle b/src/test/resources/projects/reproducible/build.gradle
index 4bc4926136fa448fd4268d7392217bf7f160fb9a..30c027462766555d41faaea0257b705a0ae58838 100644
--- a/src/test/resources/projects/reproducible/build.gradle
+++ b/src/test/resources/projects/reproducible/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index a8e1f6bb073f57d6196eebe782836ed3f436b19b..9fe3374fef22b370c6810274650d5be8b6c664b0 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 loom {
diff --git a/src/test/resources/projects/signed/build.gradle b/src/test/resources/projects/signed/build.gradle
index 45a2a278a761e5f9cc0f56bb7972719196660cfe..f4097c5e1408e35b85019fc5ed846b7c24efe9ce 100644
--- a/src/test/resources/projects/signed/build.gradle
+++ b/src/test/resources/projects/signed/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 	id 'signing'
 }
diff --git a/src/test/resources/projects/simple/build.gradle b/src/test/resources/projects/simple/build.gradle
index 5ce232ea8a8d7975252e63da3bb7a471e22c39b3..da085453d98704b87a0533934bb7fe566e76065b 100644
--- a/src/test/resources/projects/simple/build.gradle
+++ b/src/test/resources/projects/simple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/splitSources/build.gradle b/src/test/resources/projects/splitSources/build.gradle
index c4d5174f802499c7e6a6bddd3b1c7f45ea542fc7..e7047864a131cf1a53e09a6e1c3e137bfdde1c73 100644
--- a/src/test/resources/projects/splitSources/build.gradle
+++ b/src/test/resources/projects/splitSources/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.12.local'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/transitiveAccesswidener/build.gradle b/src/test/resources/projects/transitiveAccesswidener/build.gradle
index 52f1fc39f472e431795ddfe94cd85c6a01eade48..b374f3b02da2f1952df09e75759a2405d10c7c68 100644
--- a/src/test/resources/projects/transitiveAccesswidener/build.gradle
+++ b/src/test/resources/projects/transitiveAccesswidener/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/unpick/build.gradle b/src/test/resources/projects/unpick/build.gradle
index 4b79a1b01f6d59fea5174e912c213a1ee597327f..4dfb74262b0c36b9601a501bba2700f62c726725 100644
--- a/src/test/resources/projects/unpick/build.gradle
+++ b/src/test/resources/projects/unpick/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
