From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 20:46:20 -0700
Subject: [PATCH] Remove annoying warning


diff --git a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/mojmap/MojangMappingLayer.java b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/mojmap/MojangMappingLayer.java
index 541fd489e57c300ae8d7e8ade033d9b607fc3467..12c0fbedd6c83a9ec6f8643afff3a660a4d109a6 100644
--- a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/mojmap/MojangMappingLayer.java
+++ b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/mojmap/MojangMappingLayer.java
@@ -58,8 +58,6 @@ public record MojangMappingLayer(MinecraftVersionMeta.Download clientDownload,
 
 		download(clientMappings, serverMappings);
 
-		printMappingsLicense(clientMappings);
-
 		// Filter out field names matching the pattern
 		DstNameFilterMappingVisitor nameFilter = new DstNameFilterMappingVisitor(mappingVisitor, SYNTHETIC_NAME_PATTERN);
 
@@ -78,24 +76,6 @@ public record MojangMappingLayer(MinecraftVersionMeta.Download clientDownload,
 		HashedDownloadUtil.downloadIfInvalid(new URL(serverDownload().url()), serverMappings.toFile(), serverDownload().sha1(), logger(), false);
 	}
 
-	private void printMappingsLicense(Path clientMappings) {
-		try (BufferedReader clientBufferedReader = Files.newBufferedReader(clientMappings, StandardCharsets.UTF_8)) {
-			logger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-			logger().warn("Using of the official minecraft mappings is at your own risk!");
-			logger().warn("Please make sure to read and understand the following license:");
-			logger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-			String line;
-
-			while ((line = clientBufferedReader.readLine()).startsWith("#")) {
-				logger().warn(line);
-			}
-
-			logger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-		} catch (IOException e) {
-			throw new RuntimeException("Failed to read client mappings", e);
-		}
-	}
-
 	@Override
 	public MappingsNamespace getSourceNamespace() {
 		return MappingsNamespace.OFFICIAL;
