From dfeccafa93396c4ef61cd53815e00cd3d611b7d4 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:13:44 -0700
Subject: [PATCH] Use Quilt's tiny-remapper


diff --git a/build.gradle b/build.gradle
index fa1a808..d1ca807 100644
--- a/build.gradle
+++ b/build.gradle
@@ -38,6 +38,7 @@ repositories {
 		name = 'Fabric'
 		url = 'https://maven.fabricmc.net/'
 	}
+	maven { url = "https://maven.quiltmc.org/repository/release/" } // quiet
 	mavenCentral()
 }
 
@@ -78,7 +79,7 @@ dependencies {
 	}
 
 	// tinyfile management
-	implementation ('net.fabricmc:tiny-remapper:0.3.2')
+	implementation ("org.quiltmc:tiny-remapper:0.4.1") // quiet
 	implementation ('net.fabricmc:tiny-mappings-parser:0.3.0+build.17')
 
 	implementation 'net.fabricmc:access-widener:1.0.0'
-- 
2.32.0

