From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 22:01:14 -0700
Subject: [PATCH] Decompiler changes


diff --git a/build.gradle b/build.gradle
index 260fd12c8bc07ea5d38ef343af591ed44744b5d7..bb2e99549a17c7e1a225666c172b72a472e6a153 100644
--- a/build.gradle
+++ b/build.gradle
@@ -48,6 +48,12 @@ repositories {
 		name = 'Fabric'
 		url = 'https://maven.fabricmc.net/'
 	}
+	// quiet start
+	maven {
+		url = "https://maven.quiltmc.org/repository/release/"
+		mavenContent { includeGroup("org.quiltmc") }
+	}
+	// quiet end
 	mavenCentral()
 }
 
@@ -97,7 +103,11 @@ dependencies {
 	}
 
 	// decompilers
-	implementation ('net.fabricmc:fabric-fernflower:1.5.0')
+	// quiet start
+	implementation("org.quiltmc:quiltflower:1.8.1") {
+		transitive = false
+	}
+	// quiet end
 	implementation ('net.fabricmc:cfr:0.1.1')
 
 	// source code remapping
diff --git a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
index 782da7096a0eb1a9b604e77adf1a4fe1fb24a243..5f01a1ca8a35177c467454e6260b7ac00bf9e9d3 100644
--- a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
@@ -80,7 +80,7 @@ public class SingleJarDecompileConfiguration extends DecompileConfiguration<Mapp
 			task.setDescription("Decompile minecraft using the default decompiler.");
 			task.setGroup(Constants.TaskGroup.FABRIC);
 
-			task.dependsOn(project.getTasks().named("genSourcesWithCfr"));
+			task.dependsOn(project.getTasks().named("genSourcesWithQuiltflower")); // quiet
 		});
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java b/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
index 8bc63e20695c4f1fb0eca6a79c1bef3c25076eda..d7e0a254198f1a75a007a18b22d6b0de11ec23f5 100644
--- a/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
@@ -36,7 +36,7 @@ public final class DecompilerConfiguration {
 	}
 
 	public static void setup(Project project) {
-		registerDecompiler(project, "fernFlower", FabricFernFlowerDecompiler.class);
+		registerDecompiler(project, "quiltflower", FabricFernFlowerDecompiler.class); // quiet
 		registerDecompiler(project, "cfr", LoomCFRDecompiler.class);
 	}
 
diff --git a/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java b/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
index 2d2041b6fea5697d72c4c2c8ff3b43fe47101fa1..0021e41e01ee5b839212d15c5ea2c6209dd419ce 100644
--- a/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
+++ b/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
@@ -46,7 +46,7 @@ public final class FabricFernFlowerDecompiler implements LoomDecompiler {
 					IFernflowerPreferences.REMOVE_SYNTHETIC, "1",
 					IFernflowerPreferences.LOG_LEVEL, "trace",
 					IFernflowerPreferences.THREADS, String.valueOf(metaData.numberOfThreads()),
-					IFernflowerPreferences.INDENT_STRING, "\t",
+					IFernflowerPreferences.INDENT_STRING, "  ", // quiet
 					IFabricJavadocProvider.PROPERTY_NAME, new TinyJavadocProvider(metaData.javaDocs().toFile())
 				)
 		);
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
index 9e93097acaffa47a609eefe8d12a6641ab5c32af..745ab836c58aacda2bea7ae7f25b589a2786d867 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
@@ -45,7 +45,7 @@ class DecompileTest extends Specification implements GradleProjectTestTrait {
 
 		where:
 			decompiler 		| task								| version
-			'fernflower'	| "genSourcesWithFernFlower"		| PRE_RELEASE_GRADLE
+			'quiltflower'	| "genSourcesWithQuiltflower"		| PRE_RELEASE_GRADLE // quiet
 			'cfr' 			| "genSourcesWithCfr"				| PRE_RELEASE_GRADLE
 	}
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
index 99cfba0e60f3cbc8cffec0024efaf25768b40410..078c28ec4f5776f8d7a719d8ca8cc6f9b378a582 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
@@ -51,7 +51,7 @@ class FabricAPITest extends Specification implements GradleProjectTestTrait {
 
 			// Set the version to something constant
 			gradle.buildGradle.text = gradle.buildGradle.text.replace('project.version + "+" + (ENV.GITHUB_RUN_NUMBER ? "" : "local-") + getBranch()', "\"$API_VERSION\"")
-					.replace("fabric-loom", "quiet-fabric-loom") // quiet
+					.replace("fabric-loom", "quiet-fabric-loom").replace("FernFlower", "Quiltflower") // quiet
 
 			def server = ServerRunner.create(gradle.projectDir, "1.19.2")
 										.withMod(gradle.getOutputFile("fabric-api-${API_VERSION}.jar"))
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index b38fc44fb5f10c691e2d01b6d1b1e97575ec5f38..2de0b93e0fa87011db4322cd849115e317cc54ab 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -13,7 +13,7 @@ loom {
 		cfr {
 			options.put("test", "value")
 		}
-		fernflower {
+		quiltflower { // quiet
 			options.put("test", "value")
 		}
 	}
