From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 22:01:14 -0700
Subject: [PATCH] Decompiler changes


diff --git a/build.gradle b/build.gradle
index 626eb8cd06b79432f542e9cd9f66d129318a259c..d9b06a1ace42ed9166002ffd8f84eb54ddc91d56 100644
--- a/build.gradle
+++ b/build.gradle
@@ -45,6 +45,12 @@ repositories {
 		name = 'Fabric'
 		url = 'https://maven.fabricmc.net/'
 	}
+	// quiet start
+	maven {
+		url = "https://maven.quiltmc.org/repository/release/"
+		mavenContent { includeGroup("org.quiltmc") }
+	}
+	// quiet end
 	mavenCentral()
 	mavenLocal()
 }
@@ -95,7 +101,11 @@ dependencies {
 	}
 
 	// decompilers
-	implementation ('net.fabricmc:fabric-fernflower:1.4.1')
+	// quiet start
+	implementation("org.quiltmc:quiltflower:1.8.1") {
+		transitive = false
+	}
+	// quiet end
 	implementation ('net.fabricmc:cfr:0.0.9')
 
 	// source code remapping
diff --git a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
index ccaf627d9952c4b8aada19042299ec16127cdef8..8faa37d44ee8ac9c244c7c33843b7a846d2c7386 100644
--- a/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/decompile/SingleJarDecompileConfiguration.java
@@ -80,7 +80,7 @@ public class SingleJarDecompileConfiguration extends DecompileConfiguration<Mapp
 			task.setDescription("Decompile minecraft using the default decompiler.");
 			task.setGroup(Constants.TaskGroup.FABRIC);
 
-			task.dependsOn(project.getTasks().named("genSourcesWithCfr"));
+			task.dependsOn(project.getTasks().named("genSourcesWithQuiltflower")); // quiet
 		});
 	}
 }
diff --git a/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java b/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
index 8bc63e20695c4f1fb0eca6a79c1bef3c25076eda..d7e0a254198f1a75a007a18b22d6b0de11ec23f5 100644
--- a/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/decompilers/DecompilerConfiguration.java
@@ -36,7 +36,7 @@ public final class DecompilerConfiguration {
 	}
 
 	public static void setup(Project project) {
-		registerDecompiler(project, "fernFlower", FabricFernFlowerDecompiler.class);
+		registerDecompiler(project, "quiltflower", FabricFernFlowerDecompiler.class); // quiet
 		registerDecompiler(project, "cfr", LoomCFRDecompiler.class);
 	}
 
diff --git a/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java b/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
index 18d55451da32a6bae3dcc33e69ba6e661de8a1bc..57f70d762d19699ee9b95033946326268f8c6a2a 100644
--- a/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
+++ b/src/main/java/net/fabricmc/loom/decompilers/fernflower/FabricFernFlowerDecompiler.java
@@ -46,7 +46,7 @@ public final class FabricFernFlowerDecompiler implements LoomDecompiler {
 					IFernflowerPreferences.REMOVE_SYNTHETIC, "1",
 					IFernflowerPreferences.LOG_LEVEL, "trace",
 					IFernflowerPreferences.THREADS, String.valueOf(metaData.numberOfThreads()),
-					IFernflowerPreferences.INDENT_STRING, "\t",
+					IFernflowerPreferences.INDENT_STRING, "  ", // quiet
 					IFabricJavadocProvider.PROPERTY_NAME, new TinyJavadocProvider(metaData.javaDocs().toFile())
 				)
 		);
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
index 9e93097acaffa47a609eefe8d12a6641ab5c32af..745ab836c58aacda2bea7ae7f25b589a2786d867 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
@@ -45,7 +45,7 @@ class DecompileTest extends Specification implements GradleProjectTestTrait {
 
 		where:
 			decompiler 		| task								| version
-			'fernflower'	| "genSourcesWithFernFlower"		| PRE_RELEASE_GRADLE
+			'quiltflower'	| "genSourcesWithQuiltflower"		| PRE_RELEASE_GRADLE // quiet
 			'cfr' 			| "genSourcesWithCfr"				| PRE_RELEASE_GRADLE
 	}
 
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
index fa5d783b912661a25cc0ce7060ac78e5c14a98fc..36c7f29ca59750a5d28533158c5bb32199942033 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
@@ -51,7 +51,7 @@ class FabricAPITest extends Specification implements GradleProjectTestTrait {
 
 			// Set the version to something constant
 			gradle.buildGradle.text = gradle.buildGradle.text.replace('project.version + "+" + (ENV.GITHUB_RUN_NUMBER ? "" : "local-") + getBranch()', "\"$API_VERSION\"")
-					.replace("fabric-loom", "quiet-fabric-loom") // quiet
+					.replace("fabric-loom", "quiet-fabric-loom").replace("FernFlower", "Quiltflower") // quiet
 
 			def server = ServerRunner.create(gradle.projectDir, "1.17.1")
 										.withMod(gradle.getOutputFile("fabric-api-${API_VERSION}.jar"))
