From da9bb7e2708f71dce743e61d1523369eade947ca Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 20:46:20 -0700
Subject: [PATCH] Remove annoying warning


diff --git a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/MojangMappingsDependency.java b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/MojangMappingsDependency.java
index 1cd80a4..b0cdb72 100644
--- a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/MojangMappingsDependency.java
+++ b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/MojangMappingsDependency.java
@@ -101,22 +101,6 @@ public class MojangMappingsDependency implements SelfResolvingDependency {
 			}
 		}
 
-		try (BufferedReader clientBufferedReader = Files.newBufferedReader(clientMappings, StandardCharsets.UTF_8)) {
-			project.getLogger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-			project.getLogger().warn("Using of the official minecraft mappings is at your own risk!");
-			project.getLogger().warn("Please make sure to read and understand the following license:");
-			project.getLogger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-			String line;
-
-			while ((line = clientBufferedReader.readLine()).startsWith("#")) {
-				project.getLogger().warn(line);
-			}
-
-			project.getLogger().warn("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
-		} catch (IOException e) {
-			throw new RuntimeException("Failed to read client mappings", e);
-		}
-
 		return Collections.singleton(mappingsFile.toFile());
 	}
 
-- 
2.32.0

