From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 22:34:03 -0700
Subject: [PATCH] Reduce log spam when sources fail to fully remap


diff --git a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
index 17fa8a52e63b8d17b7ae5e72402b47f4046f7e31..b45ef63d5c420f88836d79cf678a3f42daf92476 100644
--- a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
+++ b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
@@ -42,6 +42,7 @@ import org.gradle.api.Project;
 import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.api.logging.configuration.ShowStacktrace; // quiet
 import org.gradle.internal.logging.progress.ProgressLogger;
 import org.gradle.internal.logging.progress.ProgressLoggerFactory;
 import org.slf4j.Logger;
@@ -144,7 +145,14 @@ public class SourceRemapper {
 		try {
 			mercury.rewrite(srcPath, dstPath);
 		} catch (Exception e) {
+			// quiet start
+			final ShowStacktrace showStacktrace = this.project.getGradle().getStartParameter().getShowStacktrace();
+			if (showStacktrace == ShowStacktrace.ALWAYS || showStacktrace == ShowStacktrace.ALWAYS_FULL) {
 			project.getLogger().warn("Could not remap " + source.getName() + " fully!", e);
+			} else {
+				this.project.getLogger().warn("Could not remap {} fully!\n  Message: '{}'.\n  Run with '--stacktrace' to print the full stacktrace.", source.getName(), e.getMessage());
+			}
+			// quiet end
 		}
 
 		copyNonJavaFiles(srcPath, dstPath, project.getLogger(), source.toPath());
