From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 22:34:03 -0700
Subject: [PATCH] Reduce log spam when sources fail to fully remap


diff --git a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
index 64261ea8d0128f9fd7106a94c1d1602ff3ff7fc6..c5ee56e909630e7721bb744cdb221e57aa45b2fd 100644
--- a/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
+++ b/src/main/java/net/fabricmc/loom/util/SourceRemapper.java
@@ -37,6 +37,7 @@ import org.cadixdev.lorenz.MappingSet;
 import org.cadixdev.mercury.Mercury;
 import org.cadixdev.mercury.remapper.MercuryRemapper;
 import org.gradle.api.Project;
+import org.gradle.api.logging.configuration.ShowStacktrace; // quiet
 import org.slf4j.Logger;
 
 import net.fabricmc.loom.LoomGradleExtension;
@@ -140,7 +141,14 @@ public class SourceRemapper {
 		try {
 			mercury.rewrite(srcPath, dstPath);
 		} catch (Exception e) {
+			// quiet start
+			final ShowStacktrace showStacktrace = this.project.getGradle().getStartParameter().getShowStacktrace();
+			if (showStacktrace == ShowStacktrace.ALWAYS || showStacktrace == ShowStacktrace.ALWAYS_FULL) {
 			project.getLogger().warn("Could not remap " + source.getName() + " fully!", e);
+			} else {
+				this.project.getLogger().warn("Could not remap {} fully!\n  Message: '{}'.\n  Run with '--stacktrace' to print the full stacktrace.", source.getName(), e.getMessage());
+			}
+			// quiet end
 		}
 
 		copyNonJavaFiles(srcPath, dstPath, project.getLogger(), source.toPath());
