From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Jul 2021 18:40:04 -0700
Subject: [PATCH] Add compileClasspath as an input for sources jar remap task


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 14b5725e0a5521e3167e725758084886a8f745aa..a1756af251900be07c86048eac5dd9155b99d4f0 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -128,6 +128,12 @@ public class RemapTaskConfiguration {
 
 			task.dependsOn(sourcesJarTask);
 			task.getInputFile().convention(sourcesJarTask.getArchiveFile());
+
+			// quiet start
+			task.getInputs().files(project.getConfigurations().named("compileClasspath"))
+					.ignoreEmptyDirectories()
+					.withPropertyName("remap-classpath");
+			// quiet end
 		});
 
 		tasks.named(BasePlugin.ASSEMBLE_TASK_NAME).configure(task -> task.dependsOn(remapSourcesTask));
