From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Jul 2021 18:40:04 -0700
Subject: [PATCH] Add compileClasspath as an input for sources jar remap task


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index a0f2ef56e48b92078db911ea81c3a35e459fcf19..da5f74fb11b8f020e8e8c73c273623b8c1cf42c2 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -128,6 +128,12 @@ public class RemapTaskConfiguration {
 
 			task.dependsOn(sourcesJarTask);
 			task.getInputFile().convention(sourcesJarTask.getArchiveFile());
+
+			// quiet start
+			task.getInputs().files(project.getConfigurations().named("compileClasspath"))
+					.ignoreEmptyDirectories()
+					.withPropertyName("remap-classpath");
+			// quiet end
 		});
 
 		tasks.named(BasePlugin.ASSEMBLE_TASK_NAME).configure(task -> task.dependsOn(remapSourcesTask));
