From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Jul 2021 18:40:04 -0700
Subject: [PATCH] Add compileClasspath as an input for sources jar remap task


diff --git a/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
index efbf5a1de4c2c39a7d8fa5f18b9ca1f65965340e..eca0b323301726740011fcabf7ed6a05ede716ea 100644
--- a/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
@@ -182,6 +182,11 @@ public class RemapConfiguration {
 			remapSourcesJarTask.getInput().convention(sourcesTask.getArchiveFile());
 			remapSourcesJarTask.getOutput().convention(sourcesTask.getArchiveFile());
 			remapSourcesJarTask.dependsOn(project.getTasks().getByName(sourcesJarTaskName));
+			// quiet start
+			remapSourcesJarTask.getInputs().files(project.getConfigurations().named("compileClasspath"))
+					.ignoreEmptyDirectories()
+					.withPropertyName("remap-classpath");
+			// quiet end
 
 			if (isDefaultRemap) {
 				// Do not use lambda here, see: https://github.com/gradle/gradle/pull/17087
