From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Jul 2021 18:40:04 -0700
Subject: [PATCH] Add compileClasspath as an input for sources jar remap task


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index d1cfc0f27d17049e41a34c42edb6d836a4a8d319..a3eb7eb0afbfd08cb7038bb76bb0ca5eb3e76fa2 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -124,6 +124,12 @@ public class RemapTaskConfiguration {
 			task.dependsOn(sourcesJarTask);
 			task.getInputFile().convention(sourcesJarTask.getArchiveFile());
 
+			// quiet start
+			task.getInputs().files(project.getConfigurations().named("compileClasspath"))
+					.ignoreEmptyDirectories()
+					.withPropertyName("remap-classpath");
+			// quiet end
+
 			if (extension.getSetupRemappedVariants().get()) {
 				project.getArtifacts().add(JavaPlugin.SOURCES_ELEMENTS_CONFIGURATION_NAME, task);
 
