From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Jul 2021 18:40:04 -0700
Subject: [PATCH] Add compileClasspath as an input for sources jar remap task


diff --git a/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
index 1a9ac21cf73540591ece2d05c4f2061889e2ef39..5d650e06dc754f9c3b63c769652fd748458107c7 100644
--- a/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/RemapConfiguration.java
@@ -188,6 +188,11 @@ public class RemapConfiguration {
 			remapSourcesJarTask.getInput().convention(sourcesTask.getArchiveFile());
 			remapSourcesJarTask.getOutput().convention(sourcesTask.getArchiveFile());
 			remapSourcesJarTask.dependsOn(project.getTasks().getByName(sourcesJarTaskName));
+			// quiet start
+			remapSourcesJarTask.getInputs().files(project.getConfigurations().named("compileClasspath"))
+					.ignoreEmptyDirectories()
+					.withPropertyName("remap-classpath");
+			// quiet end
 
 			if (isDefaultRemap) {
 				// Do not use lambda here, see: https://github.com/gradle/gradle/pull/17087
