From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 22:47:45 -0700
Subject: [PATCH] Detect shadowJar


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 00ec1dc6727fc850af0b869ed9c7d11e2a3b8fa0..7375794d945f377da9014b58f29f6201a5e310d6 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -92,6 +92,21 @@ public abstract class RemapTaskConfiguration implements Runnable {
 			// task.getDestinationDirectory().set(new File(getProject().getBuildDir(), "devlibs")); // quiet - no
 		});
 
+		// quiet start
+		getProject().getPlugins().withId("com.github.johnrengelman.shadow", new org.gradle.api.Action<>() {
+			@Override
+			public void execute(final org.gradle.api.Plugin shadowPlugin) {
+				final TaskProvider<AbstractArchiveTask> shadowJar = getTasks().named("shadowJar", AbstractArchiveTask.class);
+				shadowJar.configure(new org.gradle.api.Action<>() {
+					@Override
+					public void execute(final AbstractArchiveTask shadowJarTask) {
+						shadowJarTask.getArchiveClassifier().convention("dev-all");
+					}
+				});
+				remapJarTask.getInputFile().convention(shadowJar.flatMap(AbstractArchiveTask::getArchiveFile));
+			}
+		});
+		// quiet end
 		getTasks().named(BasePlugin.ASSEMBLE_TASK_NAME).configure(task -> task.dependsOn(remapJarTask));
 
 		trySetupSourceRemapping();
