From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index b9c11f1170f9a9a048823318102079c5befb7936..279e447c676f681364aa8432d1f7f89dd0613c3c 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -85,7 +85,7 @@ public class RemapTaskConfiguration {
 				shadowJar.configure(new org.gradle.api.Action<>() {
 					@Override
 					public void execute(final AbstractArchiveTask shadowJarTask) {
-						shadowJarTask.getArchiveClassifier().convention("dev-all");
+						shadowJarTask.getArchiveClassifier().set("dev-all");
 					}
 				});
 				remapJarTask.getInputFile().convention(shadowJar.flatMap(AbstractArchiveTask::getArchiveFile));
@@ -138,7 +138,7 @@ public class RemapTaskConfiguration {
 				return;
 			}
 
-			sourcesJarTask.getArchiveClassifier().convention("dev-sources");
+			sourcesJarTask.getArchiveClassifier().set("dev-sources"); // quiet
 			// sourcesJarTask.getDestinationDirectory().set(new File(project.getBuildDir(), "devlibs")); // quiet - no
 			task.getArchiveClassifier().convention("sources");
 
