From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sat, 21 May 2022 12:10:42 -0700
Subject: [PATCH] Update testing Gradle nightly


diff --git a/src/test/groovy/net/fabricmc/loom/test/LoomTestConstants.groovy b/src/test/groovy/net/fabricmc/loom/test/LoomTestConstants.groovy
index ea5c33ec266bbece28d51a20380c494f0373d01f..09d2527d8eb263ec25f9b79a6e8bdcda06dbb44f 100644
--- a/src/test/groovy/net/fabricmc/loom/test/LoomTestConstants.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/LoomTestConstants.groovy
@@ -27,12 +27,12 @@ package net.fabricmc.loom.test
 import org.gradle.util.GradleVersion
 
 class LoomTestConstants {
-    private final static String NIGHTLY_VERSION = "7.6-20220519002827+0000"
+    private final static String NIGHTLY_VERSION = "7.6-20220714221139+0000" // quiet
     private final static boolean NIGHTLY_EXISTS = nightlyExists(NIGHTLY_VERSION)
 
     public final static String DEFAULT_GRADLE = GradleVersion.current().getVersion()
     // Tests that depend specifically on the nightly will run on the current version when the nightly is not available.
-    public final static String PRE_RELEASE_GRADLE = NIGHTLY_EXISTS ? "7.6-20220516224938+0000" : DEFAULT_GRADLE
+    public final static String PRE_RELEASE_GRADLE = NIGHTLY_EXISTS ? NIGHTLY_VERSION : DEFAULT_GRADLE // quiet
     public final static String[] STANDARD_TEST_VERSIONS = NIGHTLY_EXISTS ? [DEFAULT_GRADLE, PRE_RELEASE_GRADLE] : [DEFAULT_GRADLE]
 
     /**
