From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sat, 9 Oct 2021 13:47:46 -0700
Subject: [PATCH] Use project directory instead of root project directory for
 runs


diff --git a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
index 86e3ddfbdda7cf9683c2b517fdd9b336be860aba..d28a5655139c84840d37e3b259590377794cb433 100644
--- a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
+++ b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
@@ -178,7 +178,10 @@ public class RunConfig {
 		runConfig.configName = configName;
 		populate(project, extension, runConfig, environment, appendProjectPath);
 		runConfig.ideaModuleName = IdeaUtils.getIdeaModuleName(new SourceSetReference(sourceSet, project));
-		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + runDir;
+		// quiet start
+		final Path relativeProjectPath = project.getRootDir().toPath().relativize(project.getProjectDir().toPath());
+		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + relativeProjectPath + "/" + runDir;
+		// quiet end
 		runConfig.runDir = runDir;
 		runConfig.sourceSet = sourceSet;
 		runConfig.environment = environment;
