From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 8f5ffa16dccca8f33900fe0a332d9597b542df00..1d8a07fa237585d58353a21648c8ae8418d93577 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -103,7 +103,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				shadowJar.configure(new Action<>() {
 					@Override
 					public void execute(final AbstractArchiveTask shadowJarTask) {
-						shadowJarTask.getArchiveClassifier().convention("dev-all");
+						shadowJarTask.getArchiveClassifier().set("dev-all");
 					}
 				});
 				getTasks().named(REMAP_JAR_TASK_NAME, RemapJarTask.class).configure(new Action<>() {
@@ -183,7 +183,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				return;
 			}
 
-			sourcesJarTask.getArchiveClassifier().convention("dev-sources");
+			sourcesJarTask.getArchiveClassifier().set("dev-sources");
 			// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
 			task.getArchiveClassifier().convention("sources");
 
