From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 81d4fe0ac33403f67b70b6add3a87383b502c3e7..16cb1716710fcdf4339884f534fa779adbea73bb 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -115,7 +115,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				shadowJar.configure(new Action<>() {
 					@Override
 					public void execute(final AbstractArchiveTask shadowJarTask) {
-						shadowJarTask.getArchiveClassifier().convention("dev-all");
+						shadowJarTask.getArchiveClassifier().set("dev-all");
 					}
 				});
 				getTasks().named(REMAP_JAR_TASK_NAME, RemapJarTask.class).configure(new Action<>() {
@@ -210,7 +210,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 
 				final Jar sourcesJarTask = (Jar) sourcesTask;
 
-				sourcesJarTask.getArchiveClassifier().convention("dev-sources");
+				sourcesJarTask.getArchiveClassifier().set("dev-sources");
 				// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
 				task.getArchiveClassifier().convention("sources");
 
