From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 89b98cffeff5ec9e006742a8f289eaa799b4c874..4cd30c828d93cdf5615b37f0b2d73429a0784d8b 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -106,7 +106,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				shadowJar.configure(new Action<>() {
 					@Override
 					public void execute(final AbstractArchiveTask shadowJarTask) {
-						shadowJarTask.getArchiveClassifier().convention("dev-all");
+						shadowJarTask.getArchiveClassifier().set("dev-all");
 					}
 				});
 				getTasks().named(REMAP_JAR_TASK_NAME, RemapJarTask.class).configure(new Action<>() {
@@ -186,7 +186,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				return;
 			}
 
-			sourcesJarTask.getArchiveClassifier().convention("dev-sources");
+			sourcesJarTask.getArchiveClassifier().set("dev-sources");
 			// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
 			task.getArchiveClassifier().convention("sources");
 
