From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 30 Nov 2023 14:16:28 -0700
Subject: [PATCH] Bump Mercury


diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 139a2e2616aab25f7202b5a94fdc77d89681c948..3fdde62ede98d85f640dc78db7e4f021309b9b08 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -11,7 +11,7 @@ tiny-remapper = "1.10.24"
 access-widener = "2.1.0"
 mapping-io = "0.4.2"
 lorenz-tiny = "4.0.2"
-mercury = "0.1.2.15"
+mercury = "0.1.4.17"
 kotlinx-metadata = "0.7.0"
 
 # Plugins
