From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 16 Jan 2024 11:32:19 -0700
Subject: [PATCH] Fix Maven publishing (cherry picked from fabric-loom)


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 4cd30c828d93cdf5615b37f0b2d73429a0784d8b..1d8a07fa237585d58353a21648c8ae8418d93577 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -92,12 +92,9 @@ public abstract class RemapTaskConfiguration implements Runnable {
 			task.getIncludesClientOnlyClasses().set(getProject().provider(extension::areEnvironmentSourceSetsSplit));
 		};
 
-		if (extension.multiProjectOptimisation()) {
-			// must not be lazy to ensure that the prepare tasks get setup for other projects to depend on.
-			getTasks().create(REMAP_JAR_TASK_NAME, RemapJarTask.class, remapJarTaskAction);
-		} else {
-			getTasks().register(REMAP_JAR_TASK_NAME, RemapJarTask.class, remapJarTaskAction);
-		}
+		// must not be lazy to ensure that the prepare tasks get setup for other projects to depend on.
+		// Being lazy also breaks maven publishing, see: https://github.com/FabricMC/fabric-loom/issues/1023
+		getTasks().create(REMAP_JAR_TASK_NAME, RemapJarTask.class, remapJarTaskAction);
 		// quiet start
 		getProject().getPlugins().withId("com.github.johnrengelman.shadow", new Action<>() {
 			@Override
