From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Mon, 22 Apr 2024 13:00:32 -0700
Subject: [PATCH] Fix issues with remapping NeoForge to intermediary by not
 doing it for Mojang-mapped NeoForge projects


diff --git a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
index 6b527d0c162b7c649161c6530b91b48fdfd37afe..b3912624ba62b2d8daa11aca27381c6a7a16860f 100644
--- a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
@@ -258,8 +258,12 @@ public abstract class CompileConfiguration implements Runnable {
 
 		final var provideContext = new AbstractMappedMinecraftProvider.ProvideContext(true, extension.refreshDeps(), configContext);
 
-		extension.setIntermediaryMinecraftProvider(intermediaryMinecraftProvider);
-		intermediaryMinecraftProvider.provide(provideContext);
+		if (!(extension.isNeoForge()
+				&& mappingConfiguration instanceof net.fabricmc.loom.configuration.providers.forge.ForgeMigratedMappingConfiguration c
+				&& c.isMojangMappedProject())) {
+			extension.setIntermediaryMinecraftProvider(intermediaryMinecraftProvider);
+			intermediaryMinecraftProvider.provide(provideContext);
+		}
 
 		extension.setNamedMinecraftProvider(namedMinecraftProvider);
 		namedMinecraftProvider.provide(provideContext);
