From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 3 Dec 2023 19:07:19 -0700
Subject: [PATCH] Delete nested jars as well as the metadata to save on disk
 space


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
index 0ac4ee963caff903d175457d551b84801306c1d5..ab10a9787b7cb4cab340cdbda71a2cd512db56f1 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
@@ -124,6 +124,7 @@ public class ModProcessor {
 	private void stripNestedJars(Path path) {
 		try {
 			ZipUtils.deleteIfExists(path, "META-INF/jarjar/metadata.json");
+			ZipUtils.deleteRecursiveIfExists(path, "META-INF/jars");
 		} catch (IOException e) {
 			throw new UncheckedIOException("Failed to strip nested jars from %s".formatted(path), e);
 		}
diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
index 81464b39f6a7d12f30f6c31a6a86becdb4a03f0e..f0213bed61017cef6ed3bc12db5260bbcac5a9a5 100644
--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
@@ -191,6 +191,14 @@ public class ZipUtils {
 		}
 	}
 
+	public static void deleteRecursiveIfExists(Path zip, String path) throws IOException {
+		try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(zip, false)) {
+			if (Files.exists(fs.getPath(path))) {
+				Files.walkFileTree(fs.getPath(path), new DeletingFileVisitor());
+			}
+		}
+	}
+
 	public static int transformString(Path zip, Collection<Pair<String, UnsafeUnaryOperator<String>>> transforms) throws IOException {
 		return transformString(zip, transforms.stream());
 	}
