From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Mon, 22 Apr 2024 13:00:32 -0700
Subject: [PATCH] Fix issues with remapping NeoForge to intermediary by not
 doing it for Mojang-mapped NeoForge projects


diff --git a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
index a4f95d2891e3b780a1bc8f53987325e7b13d9a3f..93135045a5d081ff26c117678d005b25a719eb64 100644
--- a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
@@ -258,7 +258,11 @@ public abstract class CompileConfiguration implements Runnable {
 		final var provideContext = new AbstractMappedMinecraftProvider.ProvideContext(true, extension.refreshDeps(), configContext);
 
 		extension.setIntermediaryMinecraftProvider(intermediaryMinecraftProvider);
-		intermediaryMinecraftProvider.provide(provideContext);
+		if (!(extension.isNeoForge()
+				&& mappingConfiguration instanceof net.fabricmc.loom.configuration.providers.forge.FieldMigratedMappingConfiguration c
+				&& c.isMojangMappedProject())) {
+			intermediaryMinecraftProvider.provide(provideContext);
+		}
 
 		extension.setNamedMinecraftProvider(namedMinecraftProvider);
 		namedMinecraftProvider.provide(provideContext);
