From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 4 Aug 2024 15:11:40 -0700
Subject: [PATCH] Allow using Aw2At on NeoForge projects


diff --git a/src/main/java/net/fabricmc/loom/api/NeoForgeExtensionAPI.java b/src/main/java/net/fabricmc/loom/api/NeoForgeExtensionAPI.java
index 9d5474b3dfbe078bf45031aa838d861321946054..711ffc47c9f20f152e267caf1380a45c6af20dee 100644
--- a/src/main/java/net/fabricmc/loom/api/NeoForgeExtensionAPI.java
+++ b/src/main/java/net/fabricmc/loom/api/NeoForgeExtensionAPI.java
@@ -25,11 +25,30 @@
 package net.fabricmc.loom.api;
 
 import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
 
 /**
  * This is the NeoForge extension API available to build scripts.
  */
 public interface NeoForgeExtensionAPI {
+	/**
+	 * If true, {@linkplain LoomGradleExtensionAPI#getAccessWidenerPath() the project access widener file}
+	 * will be remapped to an access transformer file if set.
+	 *
+	 * @return the property
+	 */
+	Property<Boolean> getConvertAccessWideners();
+
+	/**
+	 * A set of additional access widener files that will be converted to access transformers
+	 * {@linkplain #getConvertAccessWideners() if enabled}. The files are specified as paths in jar files
+	 * (e.g. {@code path/to/my_aw.accesswidener}).
+	 *
+	 * @return the property
+	 */
+	SetProperty<String> getExtraAccessWideners();
+
 	/**
 	 * A collection of all project access transformers.
 	 * The collection should only contain AT files, and not directories or other files.
diff --git a/src/main/java/net/fabricmc/loom/extension/NeoForgeExtensionImpl.java b/src/main/java/net/fabricmc/loom/extension/NeoForgeExtensionImpl.java
index 7795978c951009e81739eeafcb3f8722176364d2..8c70df722f3c291f2de24e7cbe06799dc1bd2528 100644
--- a/src/main/java/net/fabricmc/loom/extension/NeoForgeExtensionImpl.java
+++ b/src/main/java/net/fabricmc/loom/extension/NeoForgeExtensionImpl.java
@@ -28,15 +28,31 @@ import javax.inject.Inject;
 
 import org.gradle.api.Project;
 import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
 
 import net.fabricmc.loom.api.NeoForgeExtensionAPI;
 
 public class NeoForgeExtensionImpl implements NeoForgeExtensionAPI {
 	private final ConfigurableFileCollection accessTransformers;
+	private final Property<Boolean> convertAccessWideners;
+	private final SetProperty<String> extraAccessWideners;
 
 	@Inject
 	public NeoForgeExtensionImpl(Project project) {
 		accessTransformers = project.getObjects().fileCollection();
+		convertAccessWideners = project.getObjects().property(Boolean.class).convention(false);
+		extraAccessWideners = project.getObjects().setProperty(String.class).empty();
+	}
+
+	@Override
+	public Property<Boolean> getConvertAccessWideners() {
+		return this.convertAccessWideners;
+	}
+
+	@Override
+	public SetProperty<String> getExtraAccessWideners() {
+		return this.extraAccessWideners;
 	}
 
 	@Override
diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index be4f5249b17bd1dc0b5925d269ddb1b47aa41e0e..70d825c0f534d8b8bb88ab6036ac2f3c84e7c632 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -138,11 +138,16 @@ public abstract class RemapTaskConfiguration implements Runnable {
 		trySetupSourceRemapping();
 
 		getProject().afterEvaluate(p -> {
-			if (extension.isForge()) {
-				if (PropertyUtil.getAndFinalize(extension.getForge().getConvertAccessWideners())) {
+			if (extension.isForgeLike()) {
+				if (PropertyUtil.getAndFinalize(
+						extension.isForge()
+								? extension.getForge().getConvertAccessWideners()
+								: extension.getNeoForge().getConvertAccessWideners()
+				)) {
 					Aw2At.setup(getProject(), (RemapJarTask) getTasks().getByName(REMAP_JAR_TASK_NAME));
 				}
-
+			}
+			if (extension.isForge()) {
 				Set<String> mixinConfigs = PropertyUtil.getAndFinalize(extension.getForge().getMixinConfigs());
 
 				if (!mixinConfigs.isEmpty()) {
diff --git a/src/main/java/net/fabricmc/loom/util/aw2at/Aw2At.java b/src/main/java/net/fabricmc/loom/util/aw2at/Aw2At.java
index 72662d5157d7bbc637e15777ff07a300d9cd676e..bc912dd9ee685353f2856a726c9ea56ab0e20d1c 100644
--- a/src/main/java/net/fabricmc/loom/util/aw2at/Aw2At.java
+++ b/src/main/java/net/fabricmc/loom/util/aw2at/Aw2At.java
@@ -78,7 +78,11 @@ public final class Aw2At {
 			remapJar.getAtAccessWideners().add(awName);
 		}
 
+		if (extension.isForge()) {
 		remapJar.getAtAccessWideners().addAll(extension.getForge().getExtraAccessWideners());
+		} else if (extension.isNeoForge()) {
+			remapJar.getAtAccessWideners().addAll(extension.getNeoForge().getExtraAccessWideners());
+		}
 	}
 
 	/**
