From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 16 Jun 2024 22:50:31 -0700
Subject: [PATCH] Update tiny remapper


diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 79bb21a47c9ed8bf9a8a8a3c6e28b3ae81ed2138..d8d82a438dcf8b0a50cfd53c1c377cd9cf2a07e8 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -6,7 +6,7 @@ gson = "2.10.1"
 guava = "33.0.0-jre"
 
 stitch = "0.6.2"
-tiny-remapper = "0.10.1"
+tiny-remapper = "0.10.3"
 access-widener = "2.1.0"
 mapping-io = "0.5.1"
 lorenz-tiny = "4.0.2"
