From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 22 Oct 2024 21:39:14 -0700
Subject: [PATCH] Disable loom version validation


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
index 56026a6e8db877904c3405a729d04f714e2a0ae5..0fb240554f37ed561c08739600be2f051939b138 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
@@ -102,7 +102,7 @@ public record ArtifactMetadata(boolean isFabricMod, RemapRequirements remapRequi
 				}
 
 				if (loomVersion != null && refmapRemapType == MixinRemapType.STATIC) {
-					validateLoomVersion(loomVersion, currentLoomVersion, GradleUtils.getBooleanProperty(project, Constants.Properties.IGNORE_DEPENDENCY_LOOM_VERSION_VALIDATION));
+					// validateLoomVersion(loomVersion, currentLoomVersion, GradleUtils.getBooleanProperty(project, Constants.Properties.IGNORE_DEPENDENCY_LOOM_VERSION_VALIDATION));
 				}
 
 				if (knownIndyBsmsValue != null) {
