From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 26 Nov 2024 11:04:07 -0700
Subject: [PATCH] Update for NeoForge binpatcher fork


diff --git a/src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java b/src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java
index e2b684bdddd4f05ec1c8bc4feec47c7efc0ee675..5c39247fa7128916516f5256a4480c768af66172 100644
--- a/src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java
+++ b/src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java
@@ -56,6 +56,7 @@ import dev.architectury.loom.forge.UserdevConfig;
 import dev.architectury.loom.util.MappingOption;
 import dev.architectury.loom.util.TempFiles;
 import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
 import org.gradle.api.logging.LogLevel;
 import org.gradle.api.logging.Logger;
 import org.objectweb.asm.ClassReader;
@@ -464,8 +465,9 @@ public class MinecraftPatchedProvider {
 	private void patchJars(Path clean, Path output, Path patches) {
 		ForgeToolExecutor.exec(project, spec -> {
 			UserdevConfig.BinaryPatcherConfig config = getExtension().getForgeUserdevProvider().getConfig().binpatcher();
-			spec.classpath(DependencyDownloader.download(project, config.dependency()));
-			spec.getMainClass().set("net.minecraftforge.binarypatcher.ConsoleTool");
+			final FileCollection download = DependencyDownloader.download(project, config.dependency());
+			spec.classpath(download);
+			spec.getMainClass().set(getMainClass(download));
 
 			for (String arg : config.args()) {
 				String actual = switch (arg) {
@@ -479,6 +481,40 @@ public class MinecraftPatchedProvider {
 		});
 	}
 
+	private static String getMainClass(final Iterable<File> files) {
+		String mainClass = null;
+		IOException ex = null;
+		for (File file : files) {
+			if (file.getName().endsWith(".jar")) {
+				try (final FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(file.toPath())) {
+					final Path mfPath = fs.getPath("META-INF/MANIFEST.MF");
+					if (Files.exists(mfPath)) {
+						try (final InputStream in = Files.newInputStream(mfPath)) {
+							mainClass = new Manifest(in).getMainAttributes().getValue("Main-Class");
+						}
+					}
+				} catch (final IOException e) {
+					if (ex == null) {
+						ex = e;
+					} else {
+						ex.addSuppressed(e);
+					}
+				}
+				if (mainClass != null) {
+					break;
+				}
+			}
+		}
+		if (mainClass == null) {
+			if (ex != null) {
+				throw new UncheckedIOException(ex);
+			} else {
+				throw new RuntimeException("Failed to find main class");
+			}
+		}
+		return mainClass;
+	}
+
 	private void walkFileSystems(Path source, Path target, Predicate<Path> filter, Function<FileSystem, Iterable<Path>> toWalk, FsPathConsumer action)
 			throws IOException {
 		try (FileSystemUtil.Delegate sourceFs = FileSystemUtil.getJarFileSystem(source, false);
