From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:06:34 -0700
Subject: [PATCH] Build changes & rebrand


diff --git a/build.gradle b/build.gradle
index a9e824c9f9abb7d55d1516dfa2e53e218b002fa8..03d225ba6eb7a52be00dfa1ad9674d79cf3d95f7 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,10 +5,9 @@ plugins {
 	id 'idea'
 	id 'eclipse'
 	id 'groovy'
-	id 'checkstyle'
-	id 'codenarc'
+	// quiet - remove checkstyle
 	alias(libs.plugins.kotlin)
-	alias(libs.plugins.spotless)
+	// quiet - remove checkstyle
 	alias(libs.plugins.retry)
 }
 
@@ -26,12 +25,10 @@ group = 'net.fabricmc'
 def baseVersion = '1.14'
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-	if (ENV.EXPERIMENTAL) {
-		version = baseVersion + '.0-alpha.' + ENV.BUILD_NUMBER
-	} else {
-		version = baseVersion + '.' + ENV.BUILD_NUMBER
-	}
+// quiet start
+if (ENV.GITHUB_RUN_NUMBER) {
+	version = baseVersion + '.' + ENV.GITHUB_RUN_NUMBER
+	// quiet end
 } else {
 	version = baseVersion + '.local'
 }
@@ -195,60 +192,25 @@ java {
 	targetCompatibility = JavaVersion.VERSION_21
 }
 
-spotless {
-	lineEndings = com.diffplug.spotless.LineEnding.UNIX
-
-	java {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/loom/util/DownloadUtil.java")
-	}
-
-	groovy {
-		importOrder('java', 'javax', '', 'net.fabricmc', '\\#')
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		greclipse()
-	}
-
-	groovyGradle {
-		target 'src/**/*.gradle', '*.gradle'
-		greclipse()
-	}
-
-	kotlin {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/build.gradle.kts")
-		targetExclude("src/test/resources/projects/*/**")
-		ktlint()
-	}
-}
-
-checkstyle {
-	configFile = file('checkstyle.xml')
-	toolVersion = libs.versions.checkstyle.get()
-}
-
-codenarc {
-	toolVersion = libs.versions.codenarc.get()
-	configFile = file("codenarc.groovy")
-}
+// quiet - remove checkstyle
 
 gradlePlugin {
 	plugins {
 		fabricLoom {
-			id = 'net.fabricmc.fabric-loom'
+			id = 'xyz.jpenilla.quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomNoRemapGradlePlugin'
 		}
 		fabricLoomRemap {
-			id = 'net.fabricmc.fabric-loom-remap'
+			id = 'xyz.jpenilla.quiet-fabric-loom-remap' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomRemapGradlePlugin'
 		}
 		fabricLoomLegacy {
 			// Legacy plugin id, remove in 2.0. Use `net.fabricmc.fabric-loom-remap`
-			id = 'fabric-loom'
+			id = 'quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomGradlePlugin'
 		}
 		fabricLoomCompanion {
-			id = 'net.fabricmc.fabric-loom-companion'
+			id = 'xyz.jpenilla.quiet-fabric-loom-companion' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomCompanionGradlePlugin'
 		}
 	}
@@ -322,7 +284,7 @@ publishing {
 						Node groupId = dependency.appendChild(document.createElement('groupId'))
 						groupId.setTextContent('net.fabricmc')
 						Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-						artifactId.setTextContent('fabric-loom')
+						artifactId.setTextContent('quiet-fabric-loom') // quiet
 						Node version = dependency.appendChild(document.createElement('version'))
 						version.setTextContent(project.version)
 					})
@@ -332,13 +294,13 @@ publishing {
 	}
 	repositories {
 		maven {
-			if (ENV.MAVEN_URL) {
-				url ENV.MAVEN_URL
-				credentials {
-					username ENV.MAVEN_USERNAME
-					password ENV.MAVEN_PASSWORD
-				}
+			// quiet start
+			url("https://repo.jpenilla.xyz/snapshots/")
+			credentials {
+				username = System.getenv("JMP_REPO_USER")
+				password = System.getenv("JMP_REPO_PASSWORD")
 			}
+			// quiet end
 		}
 	}
 }
diff --git a/gradle/versions.gradle b/gradle/versions.gradle
index 77f7890f5d30c7924f42c79249367f01fdf83318..98d29a89d5a63711eb7dff6b876e540acaec5303 100644
--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -22,7 +22,6 @@ def generateVersionConstants(def sourceSet, def catalogName, def sourcesName) {
 	}
 
 	sourceSet.java.srcDir task
-	spotlessGroovyGradle.dependsOn task // Not quite sure why this is needed, but it fixes a warning.
 	compileKotlin.dependsOn task
 	sourcesJar.dependsOn task
 }
diff --git a/settings.gradle b/settings.gradle
index 6b14589d1bf1551d4376ed710f3fd73e434c4580..ba0694a30a5a47529f65ba0b159c38d04ddb9c3f 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,4 +1,4 @@
-rootProject.name = name
+rootProject.name = "quiet-fabric-loom" // quiet
 
 dependencyResolutionManagement {
 	versionCatalogs {
diff --git a/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
index b3e14661bf3008d3abea20e464552a36eb3633f6..73ad96d02c551e562f222b10a6906ec0c67899e5 100644
--- a/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
@@ -32,7 +32,7 @@ import net.fabricmc.loom.task.launch.ExportClasspathTask;
 import net.fabricmc.loom.util.Constants;
 
 public class LoomCompanionGradlePlugin implements Plugin<Project> {
-	public static final String NAME = "net.fabricmc.fabric-loom-companion";
+	public static final String NAME = "net.fabricmc.quiet-fabric-loom-companion"; // quiet
 
 	@Override
 	public void apply(Project project) {
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index 1c416b3066197ac6b5a0c36b096e323faeba1198..0c6504581ddcdaa4eab8720f14e801c3c9453d79 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -51,7 +51,7 @@ import net.fabricmc.loom.util.LibraryLocationLogger;
 import net.fabricmc.loom.util.OneDrive;
 
 public class LoomGradlePlugin implements Plugin<PluginAware> {
-	public static final String NAME = "fabric-loom";
+	public static final String NAME = "quiet-fabric-loom"; // quiet
 	public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
 	public static final String LOOM_VERSION = Objects.requireNonNullElse(LoomGradlePlugin.class.getPackage().getImplementationVersion(), "0.0.0+unknown");
 
@@ -79,7 +79,7 @@ public class LoomGradlePlugin implements Plugin<PluginAware> {
 	}
 
 	private void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quiet Fabric Loom: " + LOOM_VERSION); // quiet
 
 		LibraryLocationLogger.logLibraryVersions();
 		OneDrive.verify(project);
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
index b45e65e6c1eaabf42363c2e0c44881b8e8fa9a67..f2ad6c18ad3b2f232bb034ac799ec1b9316d8ab6 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
@@ -61,6 +61,7 @@ class FabricAPITest extends Specification implements GradleProjectTestTrait {
 		def minecraftVersion = "1.21.6-pre3"
 		def server = ServerRunner.create(gradle.projectDir, minecraftVersion)
 				.withMod(gradle.getOutputFile("fabric-api-999.0.0.jar"))
+		gradle.buildGradle.write(gradle.buildGradle.text.replace("fabric-loom", "quiet-fabric-loom")) // quiet
 
 		// Test that the dependent mod can be built against the previously built fabric-api
 		def dependentMod = gradleProject(project: "minimalBase", version: version)
diff --git a/src/test/resources/projects/accesswidener/build.gradle b/src/test/resources/projects/accesswidener/build.gradle
index f9d09cb2101cf40f76b1c2ecd4cb5eaf57d8a9fc..0756a30a57f2dd33366d96d74b5098dee024c57c 100644
--- a/src/test/resources/projects/accesswidener/build.gradle
+++ b/src/test/resources/projects/accesswidener/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/compositeBuild/build.gradle b/src/test/resources/projects/compositeBuild/build.gradle
index 0fd841a28d00ae3b665f7f27c7ec44000addee8b..0f0890cd2d2211ed785664a4ba35d3504864bfdb 100644
--- a/src/test/resources/projects/compositeBuild/build.gradle
+++ b/src/test/resources/projects/compositeBuild/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index 993ac667071e4e3e2b1a8301121e817950663d61..b38fc44fb5f10c691e2d01b6d1b1e97575ec5f38 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
index e5fcc3516cdb22dd548dde381dfd112854c915d9..8bf0ef659e7a35bdd8c29b782ad7add044110a46 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
index 2fce4f13518958bb90d91589377bfdcdd8b1344e..b5c315a176edd31db45b5b7ba717436abfc49aab 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
index 2c075f03ad35f9065fc5d9e501a236ede735a016..0218448da07934bccc350f4af49af6bd9f476945 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
@@ -1,11 +1,11 @@
 pluginManagement {
 	plugins {
-		id 'fabric-loom'
+		id 'quiet-fabric-loom' // quiet
 	}
 }
 
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencyResolutionManagement {
diff --git a/src/test/resources/projects/includedJars/build.gradle b/src/test/resources/projects/includedJars/build.gradle
index 5808e9a528be3a3ec57a506048debee521615bc7..0306243e6faffc53180e0033b9a52c7b592cbc0d 100644
--- a/src/test/resources/projects/includedJars/build.gradle
+++ b/src/test/resources/projects/includedJars/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/interfaceInjection/build.gradle b/src/test/resources/projects/interfaceInjection/build.gradle
index e631fd01bbce95d0eb6c494034cfe5fa5ec781f5..de401f596ab8eaa266cde25cc18c5c80ad66fe5e 100644
--- a/src/test/resources/projects/interfaceInjection/build.gradle
+++ b/src/test/resources/projects/interfaceInjection/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/java16/build.gradle b/src/test/resources/projects/java16/build.gradle
index 018aeb560dad93d12ab7646fb1cd86c14cbc866d..0e26c2c957ff24bd7c2164a19fbcab019254d3a9 100644
--- a/src/test/resources/projects/java16/build.gradle
+++ b/src/test/resources/projects/java16/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/kotlin/build.gradle.kts b/src/test/resources/projects/kotlin/build.gradle.kts
index e52da2c34fe295ac97d5c4a879f85eafd3526db6..26f88d2960f9125fad34230c91063bff22040c93 100644
--- a/src/test/resources/projects/kotlin/build.gradle.kts
+++ b/src/test/resources/projects/kotlin/build.gradle.kts
@@ -5,7 +5,7 @@ import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 plugins {
 	kotlin("jvm") version "2.2.21"
 	kotlin("plugin.serialization") version "2.2.21"
-	id("fabric-loom")
+    id("quiet-fabric-loom") // quiet
 	`maven-publish`
 }
 
diff --git a/src/test/resources/projects/legacy/build.gradle b/src/test/resources/projects/legacy/build.gradle
index 3d41191a4c1f358140c46c758c6a4070b7a417b3..9d9866cf17f3e3ab3db2eaa978337b710a317d1f 100644
--- a/src/test/resources/projects/legacy/build.gradle
+++ b/src/test/resources/projects/legacy/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 version = project.mod_version
diff --git a/src/test/resources/projects/localFileDependency/build.gradle b/src/test/resources/projects/localFileDependency/build.gradle
index 400d8800c8b59f373a680ee108b391bd154ee75d..c8b4b2f6364236a71cdfa39c92f7ce97000498e6 100644
--- a/src/test/resources/projects/localFileDependency/build.gradle
+++ b/src/test/resources/projects/localFileDependency/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/localRuntime/build.gradle b/src/test/resources/projects/localRuntime/build.gradle
index 4851f9ec86a187483c705f04c37f9ee6cc27eeee..61cf2d98b87f6e4529902bbb4e2b4f8baffedef0 100644
--- a/src/test/resources/projects/localRuntime/build.gradle
+++ b/src/test/resources/projects/localRuntime/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/maven/build.gradle b/src/test/resources/projects/maven/build.gradle
index 102ad866692e339d4663fbb4b39a18c64fc3715b..178c484f0f9d0bb5565af437c4e5480956078d76 100644
--- a/src/test/resources/projects/maven/build.gradle
+++ b/src/test/resources/projects/maven/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mavenLibrary/build.gradle b/src/test/resources/projects/mavenLibrary/build.gradle
index 5d641ff17f290ceb9dc09e34be7dbe4a2c393399..473611ac751870adb4571aeff07db3d8d798c4ff 100644
--- a/src/test/resources/projects/mavenLibrary/build.gradle
+++ b/src/test/resources/projects/mavenLibrary/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/minimalBase/build.gradle b/src/test/resources/projects/minimalBase/build.gradle
index 35f137b653f4806d09c3f3e7877c801a2c5c2c88..978fe9e13c75012eed4aecdb5d77fce6dc462759 100644
--- a/src/test/resources/projects/minimalBase/build.gradle
+++ b/src/test/resources/projects/minimalBase/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'net.fabricmc.fabric-loom-remap'
+	id 'xyz.jpenilla.quiet-fabric-loom-remap' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApAutoRefmap/build.gradle b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
index 62db666d9c9f8cdfd0d2980cfa1e9e9307c72df0..5dd91c518e9acba2b7c8975abafac587cd4dac78 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/build.gradle
+++ b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index 9f565cdf4d2a272b1187d6f2c89dac783fc892c9..9d3312c2af17178ab04a497403e31e456833b35c 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/modJavadoc/build.gradle b/src/test/resources/projects/modJavadoc/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..974e5b68c3802e744d72eb6f07cab634100e9bbf 100644
--- a/src/test/resources/projects/modJavadoc/build.gradle
+++ b/src/test/resources/projects/modJavadoc/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mojangMappings/build.gradle b/src/test/resources/projects/mojangMappings/build.gradle
index 8fbb2c9a02cbff335fbdb7bf8fcfc661ce69f699..efb7c370931311a349a63e9f36dc72f86e68fcee 100644
--- a/src/test/resources/projects/mojangMappings/build.gradle
+++ b/src/test/resources/projects/mojangMappings/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/multi-mc-versions/settings.gradle b/src/test/resources/projects/multi-mc-versions/settings.gradle
index ba144ea6ca6087ab819ba1a4f4ba8562d428a249..67f1beabc2f55fbc4205559606b5828003ef92de 100644
--- a/src/test/resources/projects/multi-mc-versions/settings.gradle
+++ b/src/test/resources/projects/multi-mc-versions/settings.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' apply false
+	id 'quiet-fabric-loom' apply false
 }
 
 gradle.lifecycle.beforeProject {
@@ -7,7 +7,7 @@ gradle.lifecycle.beforeProject {
 		return
 	}
 
-	apply plugin: 'fabric-loom'
+	apply plugin: 'quiet-fabric-loom'
 	apply plugin: 'loom-test-plugin'
 }
 
diff --git a/src/test/resources/projects/multiproject/build.gradle b/src/test/resources/projects/multiproject/build.gradle
index 05237cc9dc6bc5630f3f2073f34ea0dfcbc0a725..a09d1e879a5f872084f614853c85a23376c36d50 100644
--- a/src/test/resources/projects/multiproject/build.gradle
+++ b/src/test/resources/projects/multiproject/build.gradle
@@ -1,13 +1,13 @@
 plugins {
 	id "java"
 	id "java-library"
-	id 'fabric-loom' apply false
+	id 'quiet-fabric-loom' apply false // quiet
 }
 
 allprojects {
 	if (it.path == ":javalib") return;
 
-	apply plugin: "fabric-loom"
+	apply plugin: "quiet-fabric-loom" // quiet
 
 	version = "1.0.0"
 
diff --git a/src/test/resources/projects/parchment/build.gradle b/src/test/resources/projects/parchment/build.gradle
index 5fee81c5d1e07de0dfcf615795ec818a8e4705e2..b2f68f83de8a72fd9be20681c788eb1efc6befb2 100644
--- a/src/test/resources/projects/parchment/build.gradle
+++ b/src/test/resources/projects/parchment/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/reproducible/build.gradle b/src/test/resources/projects/reproducible/build.gradle
index d7f4a7943512e178350292921bffaa0af6eeed26..5d20f3e204647f5195a9752458272c064e7852a8 100644
--- a/src/test/resources/projects/reproducible/build.gradle
+++ b/src/test/resources/projects/reproducible/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index bc59e2d4ea635c2cddcc43d66a97fb0396ac3728..c13920990b176dd5dcaff43a1be8ca0e2fa1d29b 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 loom {
diff --git a/src/test/resources/projects/signed/build.gradle b/src/test/resources/projects/signed/build.gradle
index 8b2926f98894df4b241f7172a5d62eec77157a33..d2012f0426e1a74224864c11aec8fe6065ef68de 100644
--- a/src/test/resources/projects/signed/build.gradle
+++ b/src/test/resources/projects/signed/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 	id 'signing'
 }
diff --git a/src/test/resources/projects/simple/build.gradle b/src/test/resources/projects/simple/build.gradle
index 2a1cae5be1c49cfc1b753cab24586686b2343805..cc0a14773f69d4281943c5907da82df5a52839e4 100644
--- a/src/test/resources/projects/simple/build.gradle
+++ b/src/test/resources/projects/simple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/splitSources/build.gradle b/src/test/resources/projects/splitSources/build.gradle
index 2999aad9448f27d64680ca2fdaa16efdd1bd6c79..b86d4fdce675a5fbdc2d75a73ba38820b9a3614b 100644
--- a/src/test/resources/projects/splitSources/build.gradle
+++ b/src/test/resources/projects/splitSources/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/transitiveAccesswidener/build.gradle b/src/test/resources/projects/transitiveAccesswidener/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..974e5b68c3802e744d72eb6f07cab634100e9bbf 100644
--- a/src/test/resources/projects/transitiveAccesswidener/build.gradle
+++ b/src/test/resources/projects/transitiveAccesswidener/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/unpick/build.gradle b/src/test/resources/projects/unpick/build.gradle
index 4b79a1b01f6d59fea5174e912c213a1ee597327f..4dfb74262b0c36b9601a501bba2700f62c726725 100644
--- a/src/test/resources/projects/unpick/build.gradle
+++ b/src/test/resources/projects/unpick/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
