From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:06:34 -0700
Subject: [PATCH] Build changes & rebrand


diff --git a/build.gradle b/build.gradle
index f1c3f01b5769bf7b1bf66a5234828020ce071f8a..b20b83c4a44eaf9d90058b6be3030b1ac8aad6b5 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,10 +5,9 @@ plugins {
 	id 'idea'
 	id 'eclipse'
 	id 'groovy'
-	id 'checkstyle'
-	id 'codenarc'
+	// quiet - remove checkstyle
 	alias(libs.plugins.kotlin)
-	alias(libs.plugins.spotless)
+	// quiet - remove checkstyle
 	alias(libs.plugins.retry)
 }
 
@@ -22,16 +21,14 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
 	}
 }
 
-group = 'net.fabricmc'
+group = 'xyz.jpenilla' // quiet
 def baseVersion = '1.17'
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-	if (ENV.EXPERIMENTAL) {
-		version = baseVersion + '.0-alpha.' + ENV.BUILD_NUMBER
-	} else {
-		version = baseVersion + '.' + ENV.BUILD_NUMBER
-	}
+// quiet start
+if (ENV.GITHUB_RUN_NUMBER) {
+	version = baseVersion + '.' + ENV.GITHUB_RUN_NUMBER
+	// quiet end
 } else {
 	version = baseVersion + '.local'
 }
@@ -185,64 +182,29 @@ java {
 	targetCompatibility = JavaVersion.VERSION_21
 }
 
-spotless {
-	lineEndings = com.diffplug.spotless.LineEnding.UNIX
-
-	java {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/loom/util/DownloadUtil.java")
-	}
-
-	groovy {
-		importOrder('java', 'javax', '', 'net.fabricmc', '\\#')
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		greclipse()
-	}
-
-	groovyGradle {
-		target 'src/**/*.gradle', '*.gradle'
-		greclipse()
-	}
-
-	kotlin {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/build.gradle.kts")
-		targetExclude("src/test/resources/projects/*/**")
-		ktlint()
-	}
-}
-
-checkstyle {
-	configFile = file('checkstyle.xml')
-	toolVersion = libs.versions.checkstyle.get()
-}
-
-codenarc {
-	toolVersion = libs.versions.codenarc.get()
-	configFile = file("codenarc.groovy")
-}
+// quiet - remove checkstyle
 
 gradlePlugin {
 	plugins {
 		fabricLoom {
-			id = 'net.fabricmc.fabric-loom'
+			id = 'xyz.jpenilla.quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomNoRemapGradlePlugin'
 		}
 		fabricLoomRemap {
-			id = 'net.fabricmc.fabric-loom-remap'
+			id = 'xyz.jpenilla.quiet-fabric-loom-remap' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomRemapGradlePlugin'
 		}
 		fabricLoomLegacy {
 			// Legacy plugin id, remove in 2.0. Use `net.fabricmc.fabric-loom-remap`
-			id = 'fabric-loom'
+			id = 'quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomGradlePlugin'
 		}
 		fabricLoomCompanion {
-			id = 'net.fabricmc.fabric-loom-companion'
+			id = 'xyz.jpenilla.quiet-fabric-loom-companion' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomCompanionGradlePlugin'
 		}
 		fabricLoomRepositories {
-			id = 'net.fabricmc.fabric-loom-repositories'
+			id = 'xyz.jpenilla.quiet-fabric-loom-repositories' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomRepositoryPlugin'
 		}
 	}
@@ -321,9 +283,9 @@ publishing {
 						Node dependencies = root.appendChild(document.createElement('dependencies'))
 						Node dependency = dependencies.appendChild(document.createElement('dependency'))
 						Node groupId = dependency.appendChild(document.createElement('groupId'))
-						groupId.setTextContent('net.fabricmc')
+						groupId.setTextContent('xyz.jpenilla') // quiet
 						Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-						artifactId.setTextContent('fabric-loom')
+						artifactId.setTextContent('quiet-fabric-loom') // quiet
 						Node version = dependency.appendChild(document.createElement('version'))
 						version.setTextContent(markerDependencyVersion)
 					})
@@ -333,13 +295,13 @@ publishing {
 	}
 	repositories {
 		maven {
-			if (ENV.MAVEN_URL) {
-				url ENV.MAVEN_URL
-				credentials {
-					username ENV.MAVEN_USERNAME
-					password ENV.MAVEN_PASSWORD
-				}
+			// quiet start
+			url("https://repo.jpenilla.xyz/snapshots/")
+			credentials {
+				username = System.getenv("JMP_REPO_USER")
+				password = System.getenv("JMP_REPO_PASSWORD")
 			}
+			// quiet end
 		}
 	}
 }
diff --git a/gradle/versions.gradle b/gradle/versions.gradle
index 77f7890f5d30c7924f42c79249367f01fdf83318..98d29a89d5a63711eb7dff6b876e540acaec5303 100644
--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -22,7 +22,6 @@ def generateVersionConstants(def sourceSet, def catalogName, def sourcesName) {
 	}
 
 	sourceSet.java.srcDir task
-	spotlessGroovyGradle.dependsOn task // Not quite sure why this is needed, but it fixes a warning.
 	compileKotlin.dependsOn task
 	sourcesJar.dependsOn task
 }
diff --git a/settings.gradle b/settings.gradle
index 6b14589d1bf1551d4376ed710f3fd73e434c4580..ba0694a30a5a47529f65ba0b159c38d04ddb9c3f 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,4 +1,4 @@
-rootProject.name = name
+rootProject.name = "quiet-fabric-loom" // quiet
 
 dependencyResolutionManagement {
 	versionCatalogs {
diff --git a/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
index b3e14661bf3008d3abea20e464552a36eb3633f6..8b785a9845cd01f28a24930db184d2e71f52e33a 100644
--- a/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomCompanionGradlePlugin.java
@@ -32,7 +32,7 @@ import net.fabricmc.loom.task.launch.ExportClasspathTask;
 import net.fabricmc.loom.util.Constants;
 
 public class LoomCompanionGradlePlugin implements Plugin<Project> {
-	public static final String NAME = "net.fabricmc.fabric-loom-companion";
+	public static final String NAME = "xyz.jpenilla.quiet-fabric-loom-companion"; // quiet
 
 	@Override
 	public void apply(Project project) {
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index 1c416b3066197ac6b5a0c36b096e323faeba1198..0c6504581ddcdaa4eab8720f14e801c3c9453d79 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -51,7 +51,7 @@ import net.fabricmc.loom.util.LibraryLocationLogger;
 import net.fabricmc.loom.util.OneDrive;
 
 public class LoomGradlePlugin implements Plugin<PluginAware> {
-	public static final String NAME = "fabric-loom";
+	public static final String NAME = "quiet-fabric-loom"; // quiet
 	public static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
 	public static final String LOOM_VERSION = Objects.requireNonNullElse(LoomGradlePlugin.class.getPackage().getImplementationVersion(), "0.0.0+unknown");
 
@@ -79,7 +79,7 @@ public class LoomGradlePlugin implements Plugin<PluginAware> {
 	}
 
 	private void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quiet Fabric Loom: " + LOOM_VERSION); // quiet
 
 		LibraryLocationLogger.logLibraryVersions();
 		OneDrive.verify(project);
diff --git a/src/main/java/net/fabricmc/loom/LoomNoRemapGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomNoRemapGradlePlugin.java
index 05408bf6ebef3275a3a30346e1a79b047eace97e..3526e15bceb294025dc82cceacf5cf2c7ff11429 100644
--- a/src/main/java/net/fabricmc/loom/LoomNoRemapGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomNoRemapGradlePlugin.java
@@ -31,7 +31,7 @@ import org.gradle.api.Project;
  * A marker plugin to indicate to the main loom plugin not to setup for remapping.
  */
 public class LoomNoRemapGradlePlugin implements Plugin<Project> {
-	public static final String NAME = "net.fabricmc.fabric-loom";
+	public static final String NAME = "xyz.jpenilla.quiet-fabric-loom";
 
 	@Override
 	public void apply(Project target) {
diff --git a/src/main/java/net/fabricmc/loom/LoomRemapGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomRemapGradlePlugin.java
index ecdf2875b5a7930316f864fb65fa08b80a2b5edd..d7ff2ab070d4a8f9baf41cbd2343706101ab2b17 100644
--- a/src/main/java/net/fabricmc/loom/LoomRemapGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomRemapGradlePlugin.java
@@ -31,7 +31,7 @@ import org.gradle.api.Project;
  * An alias for `net.fabricmc.fabric-loom-remap` to `fabric-loom`.
  */
 public class LoomRemapGradlePlugin implements Plugin<Project> {
-	public static final String NAME = "net.fabricmc.fabric-loom-remap";
+	public static final String NAME = "xyz.jpenilla.quiet-fabric-loom-remap";
 
 	@Override
 	public void apply(Project target) {
diff --git a/src/test/resources/projects/includedJars/build.gradle b/src/test/resources/projects/includedJars/build.gradle
index 5808e9a528be3a3ec57a506048debee521615bc7..0306243e6faffc53180e0033b9a52c7b592cbc0d 100644
--- a/src/test/resources/projects/includedJars/build.gradle
+++ b/src/test/resources/projects/includedJars/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/includedJarsNoRemap/build.gradle b/src/test/resources/projects/includedJarsNoRemap/build.gradle
index 8558e5121891b8138acc1cedc67dabc1b2d11e58..68b48f7a14512840e5b208fbfda5db7c1eb48a8f 100644
--- a/src/test/resources/projects/includedJarsNoRemap/build.gradle
+++ b/src/test/resources/projects/includedJarsNoRemap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'net.fabricmc.fabric-loom'
+	id 'xyz.jpenilla.quiet-fabric-loom'
 }
 
 repositories {
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index 9f565cdf4d2a272b1187d6f2c89dac783fc892c9..9d3312c2af17178ab04a497403e31e456833b35c 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/nestJarsApi/build.gradle b/src/test/resources/projects/nestJarsApi/build.gradle
index c230b03ab5e3a61024acebe712988279fcac39ca..a71a41b4055e1b7dc7bc7ab44bfc52ab5309ed2d 100644
--- a/src/test/resources/projects/nestJarsApi/build.gradle
+++ b/src/test/resources/projects/nestJarsApi/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index 130f265a70ba8ea029a666ee69466e0cdd6c464a..812b89b76b63fad4c661556b8fb196551cc4d832 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 loom {
