From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 21:06:34 -0700
Subject: [PATCH] Build changes & rebrand


diff --git a/build.gradle b/build.gradle
index e49795dd2da198d8853532d6b08f87dd3ba90403..56ec3d94d9ce0cc0a052c7f71702b74ef7122d2a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,10 +5,9 @@ plugins {
 	id 'idea'
 	id 'eclipse'
 	id 'groovy'
-	id 'checkstyle'
-	id 'codenarc'
+	// quiet - remove checkstyle
 	alias(libs.plugins.kotlin)
-	alias(libs.plugins.spotless)
+	// quiet - remove checkstyle
 	alias(libs.plugins.retry)
 }
 
@@ -26,12 +25,10 @@ group = 'net.fabricmc'
 def baseVersion = '1.11'
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-	if (ENV.EXPERIMENTAL) {
-		version = baseVersion + '.0-alpha.' + ENV.BUILD_NUMBER
-	} else {
-		version = baseVersion + '.' + ENV.BUILD_NUMBER
-	}
+// quiet start
+if (ENV.GITHUB_RUN_NUMBER) {
+	version = baseVersion + '.' + ENV.GITHUB_RUN_NUMBER
+	// quiet end
 } else {
 	version = baseVersion + '.local'
 }
@@ -195,54 +192,12 @@ java {
 	targetCompatibility = JavaVersion.VERSION_21
 }
 
-spotless {
-	lineEndings = com.diffplug.spotless.LineEnding.UNIX
-
-	java {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/loom/util/DownloadUtil.java")
-	}
-
-	groovy {
-		importOrder('java', 'javax', '', 'net.fabricmc', '\\#')
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		greclipse()
-	}
-
-	groovyGradle {
-		target 'src/**/*.gradle', '*.gradle'
-		greclipse()
-	}
-
-	kotlin {
-		licenseHeaderFile(rootProject.file("HEADER")).yearSeparator("-")
-		targetExclude("**/build.gradle.kts")
-		targetExclude("src/test/resources/projects/*/**")
-		ktlint()
-	}
-}
-
-checkstyle {
-	configFile = file('checkstyle.xml')
-	toolVersion = libs.versions.checkstyle.get()
-}
-
-// Workaround https://github.com/gradle/gradle/issues/27035
-configurations.checkstyle {
-	resolutionStrategy.capabilitiesResolution.withCapability("com.google.collections:google-collections") {
-		select("com.google.guava:guava:0")
-	}
-}
-
-codenarc {
-	toolVersion = libs.versions.codenarc.get()
-	configFile = file("codenarc.groovy")
-}
+// quiet - remove checkstyle
 
 gradlePlugin {
 	plugins {
 		fabricLoom {
-			id = 'fabric-loom'
+			id = 'quiet-fabric-loom' // quiet
 			implementationClass = 'net.fabricmc.loom.LoomGradlePlugin'
 		}
 	}
@@ -302,8 +257,10 @@ publishing {
 
 			// Manually crate the plugin marker for snapshot versions
 			snapshotPlugin(MavenPublication) { publication ->
-				groupId = 'fabric-loom'
-				artifactId = 'fabric-loom.gradle.plugin'
+				// quiet start
+				groupId = 'quiet-fabric-loom'
+				artifactId = 'quiet-fabric-loom.gradle.plugin'
+				// quiet end
 				version = baseVersion + '-SNAPSHOT'
 
 				pom.withXml({
@@ -315,7 +272,7 @@ publishing {
 					Node groupId = dependency.appendChild(document.createElement('groupId'))
 					groupId.setTextContent('net.fabricmc')
 					Node artifactId = dependency.appendChild(document.createElement('artifactId'))
-					artifactId.setTextContent('fabric-loom')
+					artifactId.setTextContent('quiet-fabric-loom') // quiet
 					Node version = dependency.appendChild(document.createElement('version'))
 					version.setTextContent(baseVersion + '-SNAPSHOT')
 				})
@@ -324,13 +281,13 @@ publishing {
 	}
 	repositories {
 		maven {
-			if (ENV.MAVEN_URL) {
-				url ENV.MAVEN_URL
-				credentials {
-					username ENV.MAVEN_USERNAME
-					password ENV.MAVEN_PASSWORD
-				}
+			// quiet start
+			url("https://repo.jpenilla.xyz/snapshots/")
+			credentials {
+				username = System.getenv("JMP_REPO_USER")
+				password = System.getenv("JMP_REPO_PASSWORD")
 			}
+			// quiet end
 		}
 	}
 }
diff --git a/gradle/versions.gradle b/gradle/versions.gradle
index 77f7890f5d30c7924f42c79249367f01fdf83318..98d29a89d5a63711eb7dff6b876e540acaec5303 100644
--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -22,7 +22,6 @@ def generateVersionConstants(def sourceSet, def catalogName, def sourcesName) {
 	}
 
 	sourceSet.java.srcDir task
-	spotlessGroovyGradle.dependsOn task // Not quite sure why this is needed, but it fixes a warning.
 	compileKotlin.dependsOn task
 	sourcesJar.dependsOn task
 }
diff --git a/settings.gradle b/settings.gradle
index 6b14589d1bf1551d4376ed710f3fd73e434c4580..ba0694a30a5a47529f65ba0b159c38d04ddb9c3f 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,4 +1,4 @@
-rootProject.name = name
+rootProject.name = "quiet-fabric-loom" // quiet
 
 dependencyResolutionManagement {
 	versionCatalogs {
diff --git a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
index c253d5ea95187f25175393a3062a3e3bf5866070..b5ae4266c581482f51746db21d6e5119b79bf610 100644
--- a/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
+++ b/src/main/java/net/fabricmc/loom/LoomGradlePlugin.java
@@ -77,7 +77,7 @@ public class LoomGradlePlugin implements Plugin<PluginAware> {
 	}
 
 	private void apply(Project project) {
-		project.getLogger().lifecycle("Fabric Loom: " + LOOM_VERSION);
+		project.getLogger().lifecycle("Quiet Fabric Loom: " + LOOM_VERSION); // quiet
 
 		LibraryLocationLogger.logLibraryVersions();
 
diff --git a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
index 15685104b2e5d1ccf9ab27da6dce5400e441e989..dfc758756707c2f3bfc3282d146a1f9daff9d75f 100644
--- a/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/gradle/GradleUtils.java
@@ -59,7 +59,7 @@ public final class GradleUtils {
 	}
 
 	public static boolean isLoomProject(Project project) {
-		return project.getPluginManager().hasPlugin("fabric-loom");
+		return project.getPluginManager().hasPlugin("quiet-fabric-loom"); // quiet
 	}
 
 	public static Provider<Boolean> getBooleanPropertyProvider(Project project, String key) {
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
index 86b5364a09eda25ab1d01ad0f19012e27db79d64..cae2662417b8c77200acbe48fda3603670a346d9 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/FabricAPITest.groovy
@@ -61,6 +61,7 @@ class FabricAPITest extends Specification implements GradleProjectTestTrait {
 		def minecraftVersion = "1.21.6-pre3"
 		def server = ServerRunner.create(gradle.projectDir, minecraftVersion)
 				.withMod(gradle.getOutputFile("fabric-api-999.0.0.jar"))
+		gradle.buildGradle.write(gradle.buildGradle.text.replace("fabric-loom", "quiet-fabric-loom")) // quiet
 
 		// Test that the dependent mod can be built against the previously built fabric-api
 		def dependentMod = gradleProject(project: "minimalBase", version: version)
diff --git a/src/test/resources/projects/accesswidener/build.gradle b/src/test/resources/projects/accesswidener/build.gradle
index f9d09cb2101cf40f76b1c2ecd4cb5eaf57d8a9fc..0756a30a57f2dd33366d96d74b5098dee024c57c 100644
--- a/src/test/resources/projects/accesswidener/build.gradle
+++ b/src/test/resources/projects/accesswidener/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/compositeBuild/build.gradle b/src/test/resources/projects/compositeBuild/build.gradle
index 0fd841a28d00ae3b665f7f27c7ec44000addee8b..0f0890cd2d2211ed785664a4ba35d3504864bfdb 100644
--- a/src/test/resources/projects/compositeBuild/build.gradle
+++ b/src/test/resources/projects/compositeBuild/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/decompile/build.gradle b/src/test/resources/projects/decompile/build.gradle
index 993ac667071e4e3e2b1a8301121e817950663d61..b38fc44fb5f10c691e2d01b6d1b1e97575ec5f38 100644
--- a/src/test/resources/projects/decompile/build.gradle
+++ b/src/test/resources/projects/decompile/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
diff --git a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
index e5fcc3516cdb22dd548dde381dfd112854c915d9..8bf0ef659e7a35bdd8c29b782ad7add044110a46 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/basic/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
index 2fce4f13518958bb90d91589377bfdcdd8b1344e..b5c315a176edd31db45b5b7ba717436abfc49aab 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/projmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
index 2c075f03ad35f9065fc5d9e501a236ede735a016..0218448da07934bccc350f4af49af6bd9f476945 100644
--- a/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
+++ b/src/test/resources/projects/dependencyResolutionManagement/settings.gradle
@@ -1,11 +1,11 @@
 pluginManagement {
 	plugins {
-		id 'fabric-loom'
+		id 'quiet-fabric-loom' // quiet
 	}
 }
 
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencyResolutionManagement {
diff --git a/src/test/resources/projects/includedJars/build.gradle b/src/test/resources/projects/includedJars/build.gradle
index 5808e9a528be3a3ec57a506048debee521615bc7..0306243e6faffc53180e0033b9a52c7b592cbc0d 100644
--- a/src/test/resources/projects/includedJars/build.gradle
+++ b/src/test/resources/projects/includedJars/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/interfaceInjection/build.gradle b/src/test/resources/projects/interfaceInjection/build.gradle
index e631fd01bbce95d0eb6c494034cfe5fa5ec781f5..de401f596ab8eaa266cde25cc18c5c80ad66fe5e 100644
--- a/src/test/resources/projects/interfaceInjection/build.gradle
+++ b/src/test/resources/projects/interfaceInjection/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/java16/build.gradle b/src/test/resources/projects/java16/build.gradle
index 018aeb560dad93d12ab7646fb1cd86c14cbc866d..0e26c2c957ff24bd7c2164a19fbcab019254d3a9 100644
--- a/src/test/resources/projects/java16/build.gradle
+++ b/src/test/resources/projects/java16/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/kotlin/build.gradle.kts b/src/test/resources/projects/kotlin/build.gradle.kts
index 37f00bb8c6a8e08671b92f16d0818ac3b760c571..13fb2f6c6f60b667d87d9f22ce984189477549e2 100644
--- a/src/test/resources/projects/kotlin/build.gradle.kts
+++ b/src/test/resources/projects/kotlin/build.gradle.kts
@@ -5,7 +5,7 @@ import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions
 plugins {
 	kotlin("jvm") version "2.0.21"
 	kotlin("plugin.serialization") version "2.0.21"
-	id("fabric-loom")
+	id("quiet-fabric-loom") // quiet
 	`maven-publish`
 }
 
diff --git a/src/test/resources/projects/legacy/build.gradle b/src/test/resources/projects/legacy/build.gradle
index 3d41191a4c1f358140c46c758c6a4070b7a417b3..9d9866cf17f3e3ab3db2eaa978337b710a317d1f 100644
--- a/src/test/resources/projects/legacy/build.gradle
+++ b/src/test/resources/projects/legacy/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 version = project.mod_version
diff --git a/src/test/resources/projects/localFileDependency/build.gradle b/src/test/resources/projects/localFileDependency/build.gradle
index 3aa822f38d6b50225ddc8920cd945b09aef14fc6..550a6b2d8c479b37e45e2cbf72db51e42c1d2009 100644
--- a/src/test/resources/projects/localFileDependency/build.gradle
+++ b/src/test/resources/projects/localFileDependency/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/localRuntime/build.gradle b/src/test/resources/projects/localRuntime/build.gradle
index 4851f9ec86a187483c705f04c37f9ee6cc27eeee..61cf2d98b87f6e4529902bbb4e2b4f8baffedef0 100644
--- a/src/test/resources/projects/localRuntime/build.gradle
+++ b/src/test/resources/projects/localRuntime/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/maven/build.gradle b/src/test/resources/projects/maven/build.gradle
index 102ad866692e339d4663fbb4b39a18c64fc3715b..178c484f0f9d0bb5565af437c4e5480956078d76 100644
--- a/src/test/resources/projects/maven/build.gradle
+++ b/src/test/resources/projects/maven/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mavenLibrary/build.gradle b/src/test/resources/projects/mavenLibrary/build.gradle
index 5d641ff17f290ceb9dc09e34be7dbe4a2c393399..473611ac751870adb4571aeff07db3d8d798c4ff 100644
--- a/src/test/resources/projects/mavenLibrary/build.gradle
+++ b/src/test/resources/projects/mavenLibrary/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/minimalBase/build.gradle b/src/test/resources/projects/minimalBase/build.gradle
index f3dc73941bc7331e8292466b924f4a769ae55a01..8fdbe5cd5ba9db93ba06dfd2ffe4f5df86cadd29 100644
--- a/src/test/resources/projects/minimalBase/build.gradle
+++ b/src/test/resources/projects/minimalBase/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApAutoRefmap/build.gradle b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
index 62db666d9c9f8cdfd0d2980cfa1e9e9307c72df0..5dd91c518e9acba2b7c8975abafac587cd4dac78 100644
--- a/src/test/resources/projects/mixinApAutoRefmap/build.gradle
+++ b/src/test/resources/projects/mixinApAutoRefmap/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mixinApSimple/build.gradle b/src/test/resources/projects/mixinApSimple/build.gradle
index 9f565cdf4d2a272b1187d6f2c89dac783fc892c9..9d3312c2af17178ab04a497403e31e456833b35c 100644
--- a/src/test/resources/projects/mixinApSimple/build.gradle
+++ b/src/test/resources/projects/mixinApSimple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/modJavadoc/build.gradle b/src/test/resources/projects/modJavadoc/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..974e5b68c3802e744d72eb6f07cab634100e9bbf 100644
--- a/src/test/resources/projects/modJavadoc/build.gradle
+++ b/src/test/resources/projects/modJavadoc/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/mojangMappings/build.gradle b/src/test/resources/projects/mojangMappings/build.gradle
index 8fbb2c9a02cbff335fbdb7bf8fcfc661ce69f699..efb7c370931311a349a63e9f36dc72f86e68fcee 100644
--- a/src/test/resources/projects/mojangMappings/build.gradle
+++ b/src/test/resources/projects/mojangMappings/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/multi-mc-versions/settings.gradle b/src/test/resources/projects/multi-mc-versions/settings.gradle
index ba144ea6ca6087ab819ba1a4f4ba8562d428a249..67f1beabc2f55fbc4205559606b5828003ef92de 100644
--- a/src/test/resources/projects/multi-mc-versions/settings.gradle
+++ b/src/test/resources/projects/multi-mc-versions/settings.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' apply false
+	id 'quiet-fabric-loom' apply false
 }
 
 gradle.lifecycle.beforeProject {
@@ -7,7 +7,7 @@ gradle.lifecycle.beforeProject {
 		return
 	}
 
-	apply plugin: 'fabric-loom'
+	apply plugin: 'quiet-fabric-loom'
 	apply plugin: 'loom-test-plugin'
 }
 
diff --git a/src/test/resources/projects/multiproject/build.gradle b/src/test/resources/projects/multiproject/build.gradle
index ae3e637a0b52af9f00d9742c331f0cb943aaf9d4..bb46e17581f051767aacc9df45e8096486fc4468 100644
--- a/src/test/resources/projects/multiproject/build.gradle
+++ b/src/test/resources/projects/multiproject/build.gradle
@@ -1,11 +1,11 @@
 plugins {
 	id "java"
 	id "java-library"
-	id 'fabric-loom' apply false
+	id 'quiet-fabric-loom' apply false // quiet
 }
 
 allprojects {
-	apply plugin: "fabric-loom"
+	apply plugin: "quiet-fabric-loom" // quiet
 
 	version = "1.0.0"
 
diff --git a/src/test/resources/projects/parchment/build.gradle b/src/test/resources/projects/parchment/build.gradle
index 5fee81c5d1e07de0dfcf615795ec818a8e4705e2..b2f68f83de8a72fd9be20681c788eb1efc6befb2 100644
--- a/src/test/resources/projects/parchment/build.gradle
+++ b/src/test/resources/projects/parchment/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 repositories {
diff --git a/src/test/resources/projects/reproducible/build.gradle b/src/test/resources/projects/reproducible/build.gradle
index d7f4a7943512e178350292921bffaa0af6eeed26..5d20f3e204647f5195a9752458272c064e7852a8 100644
--- a/src/test/resources/projects/reproducible/build.gradle
+++ b/src/test/resources/projects/reproducible/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/runconfigs/build.gradle b/src/test/resources/projects/runconfigs/build.gradle
index bc59e2d4ea635c2cddcc43d66a97fb0396ac3728..c13920990b176dd5dcaff43a1be8ca0e2fa1d29b 100644
--- a/src/test/resources/projects/runconfigs/build.gradle
+++ b/src/test/resources/projects/runconfigs/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 loom {
diff --git a/src/test/resources/projects/signed/build.gradle b/src/test/resources/projects/signed/build.gradle
index 8b2926f98894df4b241f7172a5d62eec77157a33..d2012f0426e1a74224864c11aec8fe6065ef68de 100644
--- a/src/test/resources/projects/signed/build.gradle
+++ b/src/test/resources/projects/signed/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 	id 'signing'
 }
diff --git a/src/test/resources/projects/simple/build.gradle b/src/test/resources/projects/simple/build.gradle
index 2a1cae5be1c49cfc1b753cab24586686b2343805..cc0a14773f69d4281943c5907da82df5a52839e4 100644
--- a/src/test/resources/projects/simple/build.gradle
+++ b/src/test/resources/projects/simple/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/splitSources/build.gradle b/src/test/resources/projects/splitSources/build.gradle
index 2999aad9448f27d64680ca2fdaa16efdd1bd6c79..b86d4fdce675a5fbdc2d75a73ba38820b9a3614b 100644
--- a/src/test/resources/projects/splitSources/build.gradle
+++ b/src/test/resources/projects/splitSources/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/transitiveAccesswidener/build.gradle b/src/test/resources/projects/transitiveAccesswidener/build.gradle
index 13aeb83ec713e9d3a37cf4c38f415a502e23c4aa..974e5b68c3802e744d72eb6f07cab634100e9bbf 100644
--- a/src/test/resources/projects/transitiveAccesswidener/build.gradle
+++ b/src/test/resources/projects/transitiveAccesswidener/build.gradle
@@ -1,7 +1,7 @@
 // This is used by a range of tests that append to this file before running the gradle tasks.
 // Can be used for tests that require minimal custom setup
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 	id 'maven-publish'
 }
 
diff --git a/src/test/resources/projects/unpick/build.gradle b/src/test/resources/projects/unpick/build.gradle
index 4b79a1b01f6d59fea5174e912c213a1ee597327f..4dfb74262b0c36b9601a501bba2700f62c726725 100644
--- a/src/test/resources/projects/unpick/build.gradle
+++ b/src/test/resources/projects/unpick/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom'
+	id 'quiet-fabric-loom' // quiet
 }
 
 dependencies {
