From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 1 Jul 2021 22:01:14 -0700
Subject: [PATCH] Decompiler changes


diff --git a/src/decompilers/vineflower/net/fabricmc/loom/decompilers/vineflower/VineflowerDecompiler.java b/src/decompilers/vineflower/net/fabricmc/loom/decompilers/vineflower/VineflowerDecompiler.java
index 40f14d417772357f36fde8b66c4c557faf19a656..a256dad84ec491a0a5d2aeaaa3f423ae7cdb6380 100644
--- a/src/decompilers/vineflower/net/fabricmc/loom/decompilers/vineflower/VineflowerDecompiler.java
+++ b/src/decompilers/vineflower/net/fabricmc/loom/decompilers/vineflower/VineflowerDecompiler.java
@@ -48,7 +48,7 @@ public final class VineflowerDecompiler implements LoomInternalDecompiler {
 					IFernflowerPreferences.REMOVE_SYNTHETIC, "1",
 					IFernflowerPreferences.LOG_LEVEL, "trace",
 					IFernflowerPreferences.THREADS, String.valueOf(context.numberOfThreads()),
-					IFernflowerPreferences.INDENT_STRING, "\t"
+					IFernflowerPreferences.INDENT_STRING, "  " // quiet
 				)
 		);
 
