From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sat, 9 Oct 2021 13:47:46 -0700
Subject: [PATCH] Use project directory instead of root project directory for
 runs


diff --git a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
index 0e61bc7cbf6d4d3f71f6f4236488d151cdd55a8b..3187549229ecefa1c023ec9023e27ba6d725da5e 100644
--- a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
+++ b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
@@ -179,7 +179,10 @@ public class RunConfig {
 		runConfig.vmArgs.add("-Dfabric.dli.env=" + environment.toLowerCase());
 		runConfig.eclipseProjectName = project.getExtensions().getByType(EclipseModel.class).getProject().getName();
 		runConfig.ideaModuleName = IdeaUtils.getIdeaModuleName(new SourceSetReference(sourceSet, project));
-		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + runDir;
+		// quiet start
+		final Path relativeProjectPath = project.getRootDir().toPath().relativize(project.getProjectDir().toPath());
+		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + relativeProjectPath + "/" + runDir;
+		// quiet end
 		runConfig.runDir = runDir;
 		runConfig.sourceSet = sourceSet;
 		runConfig.environment = environment;
