From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sat, 9 Oct 2021 13:47:46 -0700
Subject: [PATCH] Use project directory instead of root project directory for
 runs


diff --git a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
index da2db05ce7695aaa9432b5484e8e0ead40200e58..ffb7a2ba103e3c9d23966395e083043c1d2c7e7a 100644
--- a/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
+++ b/src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java
@@ -143,7 +143,10 @@ public class RunConfig {
 		runConfig.vmArgs.add("-Dfabric.dli.env=" + environment.toLowerCase());
 		runConfig.eclipseProjectName = project.getExtensions().getByType(EclipseModel.class).getProject().getName();
 		runConfig.ideaModuleName = IdeaUtils.getIdeaModuleName(new SourceSetReference(sourceSet, project));
-		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + runDir;
+		// quiet start
+		final Path relativeProjectPath = project.getRootDir().toPath().relativize(project.getProjectDir().toPath());
+		runConfig.runDirIdeaUrl = "file://$PROJECT_DIR$/" + relativeProjectPath + "/" + runDir;
+		// quiet end
 		runConfig.runDir = runDir;
 		runConfig.sourceSet = sourceSet;
 		runConfig.environment = environment;
