From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 8f6e2b63b825c5aacd6c22a3738c72c067cd63ef..f5a0bdd29c82197bcc7b0998ba615004813c7b15 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -157,6 +157,12 @@ public abstract class RemapTaskConfiguration implements Runnable {
 
 			boolean finalCanRemap = canRemap;
 
+			if (canRemap) {
+				final Jar sourcesJarTask = (Jar) sourcesTask;
+				sourcesJarTask.getArchiveClassifier().set("dev-sources");
+				// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
+			}
+
 			remapSourcesTask.configure(task -> {
 				if (!finalCanRemap) {
 					task.setEnabled(false);
@@ -164,9 +170,6 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				}
 
 				final Jar sourcesJarTask = (Jar) sourcesTask;
-
-				sourcesJarTask.getArchiveClassifier().convention("dev-sources");
-				// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
 				task.getArchiveClassifier().convention("sources");
 
 				task.dependsOn(sourcesJarTask);
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/RemapOutputConfigurationTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/RemapOutputConfigurationTest.groovy
index 2236d38b8b7c4f1703ff30bcf683b29d11ecc2f7..379399aa7122c525a1311c65ce30511a8f8c7bcb 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/RemapOutputConfigurationTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/RemapOutputConfigurationTest.groovy
@@ -25,5 +25,6 @@ java {
 		result.task(":jar").outcome == SUCCESS
 		result.task(":sourcesJar").outcome == SUCCESS
 		gradle.getOutputFile("includedJars-dev.jar").exists()
+		gradle.getOutputFile("includedJars-dev-sources.jar").exists()
 	}
 }
