From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 22:47:45 -0700
Subject: [PATCH] Detect shadowJar


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 40a0bf780e105a294fcf819defe9a75e931faf8d..8e719eb3eb3fb119051d8a831099e55229ddcdd8 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -90,6 +90,26 @@ public abstract class RemapTaskConfiguration implements Runnable {
 		// must not be lazy to ensure that the prepare tasks get setup for other projects to depend on.
 		// Being lazy also breaks maven publishing, see: https://github.com/FabricMC/fabric-loom/issues/1023
 		getTasks().create(REMAP_JAR_TASK_NAME, RemapJarTask.class, remapJarTaskAction);
+		// quiet start
+		getProject().getPlugins().withId("com.github.johnrengelman.shadow", new Action<>() {
+			@Override
+			public void execute(final org.gradle.api.Plugin shadowPlugin) {
+				final TaskProvider<AbstractArchiveTask> shadowJar = getTasks().named("shadowJar", AbstractArchiveTask.class);
+				shadowJar.configure(new Action<>() {
+					@Override
+					public void execute(final AbstractArchiveTask shadowJarTask) {
+						shadowJarTask.getArchiveClassifier().convention("dev-all");
+					}
+				});
+				getTasks().named(REMAP_JAR_TASK_NAME, RemapJarTask.class).configure(new Action<>() {
+					@Override
+					public void execute(RemapJarTask remapJarTask) {
+						remapJarTask.getInputFile().convention(shadowJar.flatMap(AbstractArchiveTask::getArchiveFile));
+					}
+				});
+			}
+		});
+		// quiet end
 
 		// Configure the default jar task
 		getTasks().named(JavaPlugin.JAR_TASK_NAME, AbstractArchiveTask.class).configure(task -> {
