From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 25 Jan 2022 23:07:38 -0700
Subject: [PATCH] Use set instead of convention in some cases


diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index d23182ae2a9c1c1af90410a5d3deaea8878451d5..76b7475a7d69b81ac4943115b67320aaf30f3391 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -98,7 +98,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 				shadowJar.configure(new Action<>() {
 					@Override
 					public void execute(final AbstractArchiveTask shadowJarTask) {
-						shadowJarTask.getArchiveClassifier().convention("dev-all");
+						shadowJarTask.getArchiveClassifier().set("dev-all");
 					}
 				});
 				getTasks().named(REMAP_JAR_TASK_NAME, RemapJarTask.class).configure(new Action<>() {
@@ -175,7 +175,7 @@ public abstract class RemapTaskConfiguration implements Runnable {
 
 				final Jar sourcesJarTask = (Jar) sourcesTask;
 
-				sourcesJarTask.getArchiveClassifier().convention("dev-sources");
+				sourcesJarTask.getArchiveClassifier().set("dev-sources");
 				// sourcesJarTask.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs"))); // quiet - no
 				task.getArchiveClassifier().convention("sources");
 
