From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 28 Feb 2023 12:12:48 -0700
Subject: [PATCH] Changes to include

Add transitiveInclude configuration

diff --git a/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java b/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
index acf221ad360bb2b6a78179c8dfb8efe8269cfec2..46e4c2b2475819ff49ba7f3f67e4cc9ee1412c95 100644
--- a/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
+++ b/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
@@ -34,6 +34,7 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
@@ -44,6 +45,7 @@ import org.gradle.api.artifacts.ArtifactView;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.component.ComponentIdentifier;
 import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
+import org.gradle.api.artifacts.result.ResolvedArtifactResult;
 import org.gradle.api.artifacts.result.ResolvedVariantResult;
 import org.gradle.api.artifacts.type.ArtifactTypeDefinition;
 import org.gradle.api.file.ConfigurableFileCollection;
@@ -110,7 +112,9 @@ public abstract class NestableJarGenerationTask extends AbstractLoomTask {
 		});
 		getJars().from(artifacts.getFiles());
 		dependsOn(configuration);
-		getJarIds().set(artifacts.getArtifacts().getResolvedArtifacts().map(set -> {
+		getJarIds().putAll(artifacts.getArtifacts().getResolvedArtifacts().map(this::getIds));
+	}
+	private Map<String, Metadata> getIds(final Set<ResolvedArtifactResult> set) {
 			Map<String, Metadata> map = new HashMap<>();
 			set.forEach(artifact -> {
 				ResolvedVariantResult variant = artifact.getVariant();
@@ -159,7 +163,6 @@ public abstract class NestableJarGenerationTask extends AbstractLoomTask {
 				map.put(artifact.getFile().getName(), metadata);
 			});
 			return map;
-		}));
 	}
 
 	// Generates a barebones mod for a dependency
diff --git a/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java b/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
index 9d26544316c998dedf25699705b4e81a9d2e9a62..295febcc367a7b1b1dff5d7b88c415a5c85135b2 100644
--- a/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
+++ b/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
@@ -94,6 +94,7 @@ public abstract class LoomConfigurations implements Runnable {
 		registerNonTransitive(Constants.Configurations.MINECRAFT, Role.NONE);
 
 		Provider<Configuration> include = register(Constants.Configurations.INCLUDE, Role.NONE);
+		register(Constants.Configurations.TRANSITIVE_INCLUDE, Role.RESOLVABLE);
 		register(Constants.Configurations.INCLUDE_INTERNAL, Role.RESOLVABLE).configure(configuration -> {
 			configuration.getDependencies().addAllLater(getProject().provider(() -> {
 				List<Dependency> dependencies = new ArrayList<>();
diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 64bff85415ef8225ec097ef7aa5f7eb4a7721536..cd12c084882c8d31437c718a788d910e97f2eab7 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -73,8 +73,10 @@ public abstract class RemapTaskConfiguration implements Runnable {
 		}
 
 		Configuration includeConfiguration = getProject().getConfigurations().getByName(Constants.Configurations.INCLUDE_INTERNAL);
+		Configuration transitiveIncludeConfiguration = getProject().getConfigurations().getByName(Constants.Configurations.TRANSITIVE_INCLUDE);
 		getTasks().register(Constants.Task.PROCESS_INCLUDE_JARS, NestableJarGenerationTask.class, task -> {
 			task.from(includeConfiguration);
+			task.from(transitiveIncludeConfiguration);
 			task.getOutputDirectory().set(getProject().getLayout().getBuildDirectory().dir(task.getName()));
 		});
 
diff --git a/src/main/java/net/fabricmc/loom/util/Constants.java b/src/main/java/net/fabricmc/loom/util/Constants.java
index b03c6a21c4eeffec0298febd31ffb5670a7cc316..758d564d2ad245c61a389ae130d715d1b949d4aa 100644
--- a/src/main/java/net/fabricmc/loom/util/Constants.java
+++ b/src/main/java/net/fabricmc/loom/util/Constants.java
@@ -48,6 +48,7 @@ public class Constants {
 		public static final String MOD_COMPILE_CLASSPATH_MAPPED = "modCompileClasspathMapped";
 		public static final String INCLUDE = "include";
 		public static final String INCLUDE_INTERNAL = "includeInternal";
+		public static final String TRANSITIVE_INCLUDE = "transitiveInclude";
 		public static final String MINECRAFT = "minecraft";
 
 		public static final String MINECRAFT_COMPILE_LIBRARIES = "minecraftLibraries";
