From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 28 Feb 2023 12:12:48 -0700
Subject: [PATCH] Changes to include

Add transitiveInclude configuration

diff --git a/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java b/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
index 0b59ee19854d49e028948906f1daded325b8c15a..58d27650c566462803b68b8b682135b4b7e6b515 100644
--- a/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
+++ b/src/main/java/net/fabricmc/loom/build/nesting/NestableJarGenerationTask.java
@@ -35,6 +35,7 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
@@ -43,6 +44,7 @@ import org.gradle.api.artifacts.ArtifactView;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.component.ComponentIdentifier;
 import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
+import org.gradle.api.artifacts.result.ResolvedArtifactResult;
 import org.gradle.api.artifacts.result.ResolvedVariantResult;
 import org.gradle.api.artifacts.type.ArtifactTypeDefinition;
 import org.gradle.api.file.ConfigurableFileCollection;
@@ -111,7 +113,9 @@ public abstract class NestableJarGenerationTask extends AbstractLoomTask {
 		});
 		getJars().from(artifacts.getFiles());
 		dependsOn(configuration);
-		getJarIds().set(artifacts.getArtifacts().getResolvedArtifacts().map(set -> {
+		getJarIds().putAll(artifacts.getArtifacts().getResolvedArtifacts().map(this::getIds));
+	}
+	private Map<String, Metadata> getIds(final Set<ResolvedArtifactResult> set) {
 			Map<String, Metadata> map = new HashMap<>();
 			set.forEach(artifact -> {
 				ResolvedVariantResult variant = artifact.getVariant();
@@ -160,7 +164,6 @@ public abstract class NestableJarGenerationTask extends AbstractLoomTask {
 				map.put(artifact.getFile().getName(), metadata);
 			});
 			return map;
-		}));
 	}
 
 	// Generates a barebones mod for a dependency
diff --git a/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java b/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
index 92b166e02a5f70dcf1c6cd273fec1fbba26e2024..b221ece9632a649d02ea706822d5eacab12ca101 100644
--- a/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
+++ b/src/main/java/net/fabricmc/loom/configuration/LoomConfigurations.java
@@ -94,6 +94,7 @@ public abstract class LoomConfigurations implements Runnable {
 		registerNonTransitive(Constants.Configurations.MINECRAFT, Role.NONE);
 
 		Provider<Configuration> include = register(Constants.Configurations.INCLUDE, Role.NONE);
+		register(Constants.Configurations.TRANSITIVE_INCLUDE, Role.RESOLVABLE);
 		register(Constants.Configurations.INCLUDE_INTERNAL, Role.RESOLVABLE).configure(configuration -> {
 			configuration.getDependencies().addAllLater(getProject().provider(() -> {
 				List<Dependency> dependencies = new ArrayList<>();
diff --git a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
index 40900fa41360aa4d5685499905fa13661961e760..e29f7c49276a20a296ce7ef4f12c6562bceef30d 100644
--- a/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java
@@ -68,8 +68,10 @@ public abstract class RemapTaskConfiguration implements Runnable {
 		SyncTaskBuildService.register(getProject());
 
 		Configuration includeConfiguration = getProject().getConfigurations().getByName(Constants.Configurations.INCLUDE_INTERNAL);
+		Configuration transitiveIncludeConfiguration = getProject().getConfigurations().getByName(Constants.Configurations.TRANSITIVE_INCLUDE);
 		TaskProvider<NestableJarGenerationTask> processIncludeJarsTask = getTasks().register(Constants.Task.PROCESS_INCLUDE_JARS, NestableJarGenerationTask.class, task -> {
 			task.from(includeConfiguration);
+			task.from(transitiveIncludeConfiguration);
 			task.getOutputDirectory().set(getProject().getLayout().getBuildDirectory().dir(task.getName()));
 		});
 
diff --git a/src/main/java/net/fabricmc/loom/util/Constants.java b/src/main/java/net/fabricmc/loom/util/Constants.java
index 53fcb19c6d87eb2fc2fdabc6f47568f974682017..361e020c1017fc947e1e8b89b6a925ee1b3d4a24 100644
--- a/src/main/java/net/fabricmc/loom/util/Constants.java
+++ b/src/main/java/net/fabricmc/loom/util/Constants.java
@@ -50,6 +50,7 @@ public class Constants {
 		public static final String MOD_COMPILE_CLASSPATH_MAPPED = "modCompileClasspathMapped";
 		public static final String INCLUDE = "include";
 		public static final String INCLUDE_INTERNAL = "includeInternal";
+		public static final String TRANSITIVE_INCLUDE = "transitiveInclude";
 		public static final String MINECRAFT = "minecraft";
 
 		public static final String MINECRAFT_COMPILE_LIBRARIES = "minecraftLibraries";
