From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Wed, 15 Nov 2023 12:49:27 -0700
Subject: [PATCH] Pass LinkOptions.NOFOLLOW_LINKS for exists checks


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
index 4ee3c72d800648fadd174dd72535906183744a88..b9d38f41b322013e771a861e75c7030e03bf97af 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
@@ -28,6 +28,7 @@ import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Collections;
@@ -60,7 +61,7 @@ public record ArtifactMetadata(boolean isFabricMod, RemapRequirements remapRequi
 			isFabricMod = FabricModJsonFactory.containsMod(fs);
 			final Path manifestPath = fs.getPath(Constants.Manifest.PATH);
 
-			if (Files.exists(manifestPath)) {
+			if (Files.exists(manifestPath, LinkOption.NOFOLLOW_LINKS)) {
 				final var manifest = new Manifest(new ByteArrayInputStream(Files.readAllBytes(manifestPath)));
 				final Attributes mainAttributes = manifest.getMainAttributes();
 				final String remapValue = mainAttributes.getValue(Constants.Manifest.REMAP_KEY);
@@ -92,7 +93,7 @@ public record ArtifactMetadata(boolean isFabricMod, RemapRequirements remapRequi
 
 			final Path installerPath = fs.getPath(INSTALLER_PATH);
 
-			if (isFabricMod && Files.exists(installerPath)) {
+			if (isFabricMod && Files.exists(installerPath, LinkOption.NOFOLLOW_LINKS)) {
 				final JsonObject jsonObject = LoomGradlePlugin.GSON.fromJson(Files.readString(installerPath, StandardCharsets.UTF_8), JsonObject.class);
 				installerData = new InstallerData(artifact.version(), jsonObject);
 			}
diff --git a/src/main/java/net/fabricmc/loom/util/FileSystemUtil.java b/src/main/java/net/fabricmc/loom/util/FileSystemUtil.java
index 3d457766becaba709a1adc0401bf26e177d723c1..860ce0e9a0ce64681bf776d690c61c64a2fee171 100644
--- a/src/main/java/net/fabricmc/loom/util/FileSystemUtil.java
+++ b/src/main/java/net/fabricmc/loom/util/FileSystemUtil.java
@@ -34,6 +34,7 @@ import java.nio.file.FileSystem;
 import java.nio.file.FileSystemNotFoundException;
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.util.function.Supplier;
@@ -53,7 +54,7 @@ public final class FileSystemUtil {
 		public byte[] readAllBytes(String path) throws IOException {
 			Path fsPath = getPath(path);
 
-			if (Files.exists(fsPath)) {
+			if (Files.exists(fsPath, LinkOption.NOFOLLOW_LINKS)) {
 				return Files.readAllBytes(fsPath);
 			} else {
 				throw new NoSuchFileException(fsPath.toString());
diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
index 9d73a4c24bc02d075165bf1a6f59fb85d6c05ce2..2de4439cbc652ecff6a6873f710b84569a65da68 100644
--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
@@ -31,6 +31,7 @@ import java.io.InputStreamReader;
 import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
 import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
@@ -71,7 +72,7 @@ public class ZipUtils {
 		try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(zip, false)) {
 			Path fsPath = fs.get().getPath(path);
 
-			return Files.exists(fsPath);
+			return Files.exists(fsPath, LinkOption.NOFOLLOW_LINKS);
 		} catch (IOException e) {
 			throw new UncheckedIOException("Failed to check file from zip", e);
 		}
diff --git a/src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java b/src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java
index 0cfec9c6fdb6cba02c524d5c575c8c76a7bdafe3..be87ed857230c0484c8615d60e4142581cd58af5 100644
--- a/src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java
+++ b/src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java
@@ -32,6 +32,7 @@ import java.io.Reader;
 import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
 import java.nio.file.Path;
 import java.util.Optional;
 
@@ -142,6 +143,6 @@ public final class FabricModJsonFactory {
 	}
 
 	public static boolean containsMod(FileSystemUtil.Delegate fs) {
-		return Files.exists(fs.getPath(FABRIC_MOD_JSON));
+		return Files.exists(fs.getPath(FABRIC_MOD_JSON), LinkOption.NOFOLLOW_LINKS);
 	}
 }
