From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 3 Dec 2023 19:07:19 -0700
Subject: [PATCH] Delete nested jars as well as the metadata to save on disk
 space


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
index 28385f8b98df47926c48fafe19bf0233c8224035..eb45ebe9cc0b199d9d3bf61b00efe6ad3a4603a7 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
@@ -130,6 +130,7 @@ public class ModProcessor {
 				json.remove("jars");
 				return json;
 			}));
+			ZipUtils.deleteRecursiveIfExists(path, "META-INF/jars");
 		} catch (IOException e) {
 			throw new UncheckedIOException("Failed to strip nested jars from %s".formatted(path), e);
 		}
diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
index 533904baeb126e8ad738fc6d4b1635ea4d9ff7e4..6e8d64676a5919b4ed26ddcf91e979100575a027 100644
--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
@@ -185,6 +185,14 @@ public class ZipUtils {
 		}
 	}
 
+	public static void deleteRecursiveIfExists(Path zip, String path) throws IOException {
+		try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(zip, false)) {
+			if (Files.exists(fs.getPath(path))) {
+				Files.walkFileTree(fs.getPath(path), new DeletingFileVisitor());
+			}
+		}
+	}
+
 	public static int transformString(Path zip, Collection<Pair<String, UnsafeUnaryOperator<String>>> transforms) throws IOException {
 		return transformString(zip, transforms.stream());
 	}
