From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sat, 3 Feb 2024 17:54:42 -0700
Subject: [PATCH] Disable some decompiler tests


diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/DebugLineNumbersTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/DebugLineNumbersTest.groovy
index 4d893ea63c160d3e46e08d76d4263a134fed097a..ca3d6b31260d881c6df24328888512dc88c25f5a 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/DebugLineNumbersTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/DebugLineNumbersTest.groovy
@@ -24,6 +24,8 @@
 
 package net.fabricmc.loom.test.integration
 
+import org.junit.jupiter.api.condition.DisabledIf
+
 import java.nio.charset.StandardCharsets
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.CompletionStage
@@ -50,6 +52,7 @@ import static net.fabricmc.loom.test.LoomTestConstants.PRE_RELEASE_GRADLE
 import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
 
 @Timeout(value = 30, unit = TimeUnit.MINUTES)
+@DisabledIf("disabled")
 class DebugLineNumbersTest extends Specification implements GradleProjectTestTrait {
 	static final String MAPPINGS = "1.20.1-net.fabricmc.yarn.1_20_1.1.20.1+build.1-v2"
 	static final Map<String, Integer> BREAKPOINTS = [
@@ -58,6 +61,14 @@ class DebugLineNumbersTest extends Specification implements GradleProjectTestTra
 		"net.minecraft.registry.RegistryOps": 44
 	]
 
+	boolean disabled() {
+		def ci = System.getenv("CI")
+		if (ci != null && ci.toBoolean()) {
+			return true
+		}
+		return false
+	}
+
 	def "Debug test"() {
 		setup:
 		def gradle = gradleProject(project: "minimalBase", version: PRE_RELEASE_GRADLE)
diff --git a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
index bb084f5cc771151aefaddf49f129207d8a343138..fdb369b7f220cc05e4becd2d6bb22614f9815f26 100644
--- a/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
+++ b/src/test/groovy/net/fabricmc/loom/test/integration/DecompileTest.groovy
@@ -46,10 +46,11 @@ class DecompileTest extends Specification implements GradleProjectTestTrait {
 
 		where:
 		decompiler 		| task								| version
-		'cfr' 			| "genSourcesWithCfr"				| PRE_RELEASE_GRADLE
+		//'cfr' 			| "genSourcesWithCfr"				| PRE_RELEASE_GRADLE
 		'vineflower' 	| "genSourcesWithVineflower"		| PRE_RELEASE_GRADLE
 	}
 
+	/*
 	@Unroll
 	def "custom decompiler (gradle #version)"() {
 		setup:
@@ -73,6 +74,7 @@ class DecompileTest extends Specification implements GradleProjectTestTrait {
 		where:
 		version << STANDARD_TEST_VERSIONS
 	}
+	 */
 
 	def "decompile cache"() {
 		setup:
