From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 3 Dec 2023 19:07:19 -0700
Subject: [PATCH] Delete nested jars as well as the metadata to save on disk
 space


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
index b9e7b8cc1f03cd4bed4f70b5562b646dce60500e..9cd5c1709af4e2e8a7908003a62f34135a4d91bf 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
@@ -124,6 +124,7 @@ public class ModProcessor {
 				json.remove("jars");
 				return json;
 			}));
+			ZipUtils.deleteRecursiveIfExists(path, "META-INF/jars");
 		} catch (IOException e) {
 			throw new UncheckedIOException("Failed to strip nested jars from %s".formatted(path), e);
 		}
diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
index 2de4439cbc652ecff6a6873f710b84569a65da68..0768c5a07925f462bb045238e04f34c2cbc5c6f6 100644
--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
@@ -185,6 +185,14 @@ public class ZipUtils {
 		}
 	}
 
+	public static void deleteRecursiveIfExists(Path zip, String path) throws IOException {
+		try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(zip, false)) {
+			if (Files.exists(fs.getPath(path))) {
+				Files.walkFileTree(fs.getPath(path), new DeletingFileVisitor());
+			}
+		}
+	}
+
 	public static int transformString(Path zip, Collection<Pair<String, UnsafeUnaryOperator<String>>> transforms) throws IOException {
 		return transformString(zip, transforms.stream());
 	}
