From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Sun, 3 Dec 2023 19:07:19 -0700
Subject: [PATCH] Delete nested jars as well as the metadata to save on disk
 space


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
index 83c4bc7e26b624c835cd4735a281aa2919039c37..1b4ea085d83368d54889b7523ac6222db64b69fa 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java
@@ -124,6 +124,7 @@ public class ModProcessor {
 				json.remove("jars");
 				return json;
 			}));
+			ZipUtils.deleteRecursiveIfExists(path, "META-INF/jars");
 		} catch (IOException e) {
 			throw new UncheckedIOException("Failed to strip nested jars from %s".formatted(path), e);
 		}
diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
index 44898539e8e7b7f6a1a4de065253695bef962d12..b3e021e17a808c222bd0c736e3ce70b49ba84090 100644
--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
@@ -185,6 +185,14 @@ public class ZipUtils {
 		}
 	}
 
+	public static void deleteRecursiveIfExists(Path zip, String path) throws IOException {
+		try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(zip, false)) {
+			if (Files.exists(fs.getPath(path))) {
+				Files.walkFileTree(fs.getPath(path), new DeletingFileVisitor());
+			}
+		}
+	}
+
 	public static int transformString(Path zip, Collection<Pair<String, UnsafeUnaryOperator<String>>> transforms) throws IOException {
 		return transformString(zip, transforms.stream());
 	}
