From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Tue, 22 Oct 2024 21:39:29 -0700
Subject: [PATCH] Disable loom version validation


diff --git a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
index b9d38f41b322013e771a861e75c7030e03bf97af..bff25997646f6e2f8a01799579508d3e6787a8b5 100644
--- a/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
+++ b/src/main/java/net/fabricmc/loom/configuration/mods/ArtifactMetadata.java
@@ -83,7 +83,7 @@ public record ArtifactMetadata(boolean isFabricMod, RemapRequirements remapRequi
 				}
 
 				if (loomVersion != null && refmapRemapType == MixinRemapType.STATIC) {
-					validateLoomVersion(loomVersion, currentLoomVersion);
+					// validateLoomVersion(loomVersion, currentLoomVersion);
 				}
 
 				if (knownIndyBsmsValue != null) {
