From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 28 Nov 2024 13:31:05 -0700
Subject: [PATCH] Fix getKnownIndyBsms storing a set in a list

This caused remapJar to re-run unnecessarily

diff --git a/src/main/java/net/fabricmc/loom/task/service/TinyRemapperService.java b/src/main/java/net/fabricmc/loom/task/service/TinyRemapperService.java
index 9ef76594d6326054e0bc9b1a65eda08824d2439f..7f0ae8d136e6f04e565ac6449405ec2159dcb172 100644
--- a/src/main/java/net/fabricmc/loom/task/service/TinyRemapperService.java
+++ b/src/main/java/net/fabricmc/loom/task/service/TinyRemapperService.java
@@ -110,7 +110,7 @@ public class TinyRemapperService extends Service<TinyRemapperService.Options> im
 			options.getUselegacyMixinAP().set(legacyMixin);
 			options.getKotlinClasspathService().set(KotlinClasspathService.createOptions(project));
 			options.getClasspath().from(classpath);
-			options.getKnownIndyBsms().set(extension.getKnownIndyBsms());
+			options.getKnownIndyBsms().set(extension.getKnownIndyBsms().get().stream().sorted().toList());
 			options.getRemapperExtensions().set(extension.getRemapperExtensions());
 		});
 	}
