From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Fri, 21 Nov 2025 13:26:11 -0700
Subject: [PATCH] Loom PR 1453


diff --git a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
index fe8a640a65dcdf36b18fe50d8e32d46c20848513..7c990c7b125a970d08b272fbfa2f8bd6647450fc 100644
--- a/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java
@@ -271,7 +271,9 @@ public abstract class CompileConfiguration implements Runnable {
 		}
 
 		getProject().getTasks().named(JavaPlugin.TEST_TASK_NAME, Test.class, test -> {
-			test.getInputs().property("LoomClassPathGroups", ClasspathGroupService.create(getProject()));
+			org.gradle.api.provider.Provider<ClasspathGroupService.Options> optionsProvider = ClasspathGroupService.create(getProject());
+			test.getInputs().property("LoomClassPathGroups", optionsProvider);
+			test.getInputs().files(optionsProvider.map((ClasspathGroupService.Options::getExternalClasspathGroups)));
 			test.doFirst(new Action<Task>() {
 				@Override
 				public void execute(Task task) {
