From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jason Penilla <11360596+jpenilla@users.noreply.github.com>
Date: Thu, 2 Apr 2026 19:14:13 -0700
Subject: [PATCH] Support nested jars on shadowJar tasks


diff --git a/src/main/java/net/fabricmc/loom/task/NestJarsAction.java b/src/main/java/net/fabricmc/loom/task/NestJarsAction.java
index 0a93c7f38a748dfb63b53dff273238a3f78bd293..67dae164989bd4ddf841488f413bea0af1b75466 100644
--- a/src/main/java/net/fabricmc/loom/task/NestJarsAction.java
+++ b/src/main/java/net/fabricmc/loom/task/NestJarsAction.java
@@ -36,6 +36,7 @@ import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.api.file.FileCollection;
 import org.gradle.api.file.RegularFileProperty;
 import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
 import org.gradle.jvm.tasks.Jar;
 import org.gradle.workers.WorkAction;
 import org.gradle.workers.WorkParameters;
@@ -58,7 +59,7 @@ public abstract class NestJarsAction implements Action<Task>, Serializable {
 	@Inject
 	protected abstract WorkerExecutor getWorkerExecutor();
 
-	public static void addToTask(Jar task, FileCollection jars) {
+	public static void addToTask(AbstractArchiveTask task, FileCollection jars) {
 		NestJarsAction nestJarsAction = task.getProject().getObjects().newInstance(NestJarsAction.class);
 		nestJarsAction.getJars().from(jars);
 		task.getInputs().files(nestJarsAction.getJars()); // I don't think @InputFiles works, so to be sure add the jars to the task input anyway.
diff --git a/src/main/java/net/fabricmc/loom/task/NonRemappedJarTaskConfiguration.java b/src/main/java/net/fabricmc/loom/task/NonRemappedJarTaskConfiguration.java
index e428486939b9a46b862d633a8ba1ae2bc8408c79..a28752d747b9c6f8eb1d95865585901e42d497ab 100644
--- a/src/main/java/net/fabricmc/loom/task/NonRemappedJarTaskConfiguration.java
+++ b/src/main/java/net/fabricmc/loom/task/NonRemappedJarTaskConfiguration.java
@@ -34,6 +34,7 @@ import org.gradle.api.plugins.JavaPlugin;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
 import org.gradle.jvm.tasks.Jar;
 
 import net.fabricmc.loom.LoomGradleExtension;
@@ -62,7 +63,10 @@ public class NonRemappedJarTaskConfiguration {
 	public void configure() {
 		final Provider<JarManifestService> manifestServiceProvider = JarManifestService.get(project);
 
-		project.getTasks().named(JavaPlugin.JAR_TASK_NAME, Jar.class).configure(task -> {
+		project.getTasks().withType(AbstractArchiveTask.class).configureEach(task -> {
+			if (!task.getName().equals(JavaPlugin.JAR_TASK_NAME) && !task.getName().equals("shadowJar")) {
+				return;
+			}
 			task.dependsOn(processIncludeJarsTask);
 
 			NestJarsAction.addToTask(task, project.fileTree(processIncludeJarsTask.flatMap(NestableJarGenerationTask::getOutputDirectory))
